<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.313">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Gibran Hemani">

<title>MR of protein on disease following PRS-protein assocs</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="all_cis_mr_files/libs/clipboard/clipboard.min.js"></script>
<script src="all_cis_mr_files/libs/quarto-html/quarto.js"></script>
<script src="all_cis_mr_files/libs/quarto-html/popper.min.js"></script>
<script src="all_cis_mr_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="all_cis_mr_files/libs/quarto-html/anchor.min.js"></script>
<link href="all_cis_mr_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="all_cis_mr_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="all_cis_mr_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="all_cis_mr_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="all_cis_mr_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">MR of protein on disease following PRS-protein assocs</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Gibran Hemani </p>
          </div>
  </div>
    
  
    
  </div>
  

</header>

<p>Narrative</p>
<ul>
<li>Proteome-wide forward MR finds more assocs than restricting forward MR to those found from PRS</li>
<li>There is very little agreement between cis and trans instruments</li>
<li>Slightly more likely to find evidence from trans if there is evidence from cis</li>
<li>significant cis MR agrees with PRS, no relationship between PRS and non-sig cis MR</li>
<li>trans MR much stronger relationship with PRS, which is slightly attenuated after steiger filtering</li>
<li>most instruments pass steiger filtering but all those that don’t are trans</li>
<li>11 examples of bi-directional effects</li>
<li>todo: coloc for cis-MR</li>
</ul>
<div class="cell" data-hash="all_cis_mr_cache/html/unnamed-chunk-1_8e340af49e725eb1f09c885ad93d9724">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'dplyr'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    filter, lag</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    intersect, setdiff, setequal, union</code></pre>
</div>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ieugwasr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>API: public: http://gwas-api.mrcieu.ac.uk/</code></pre>
</div>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyr)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(TwoSampleMR)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>TwoSampleMR version 0.5.7 
[&gt;] New: Option to use non-European LD reference panels for clumping etc
[&gt;] Some studies temporarily quarantined to verify effect allele
[&gt;] See news(package='TwoSampleMR') and https://gwas.mrcieu.ac.uk for further details</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'TwoSampleMR'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:ieugwasr':

    ld_matrix</code></pre>
</div>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(here)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>here() starts at /Users/gh13047/repo/ukbb-prot-prs</code></pre>
</div>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(metafor)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: Matrix</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'Matrix'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:tidyr':

    expand, pack, unpack</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: metadat</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: numDeriv</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Loading the 'metafor' package (version 4.2-0). For an
introduction to the package please type: help(metafor)</code></pre>
</div>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-hash="all_cis_mr_cache/html/unnamed-chunk-2_3544dfb71712356188913abc092c08ea">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>traits <span class="ot">&lt;-</span> <span class="fu">read_xlsx</span>(<span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"Supplementary_Table5.xlsx"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>New names:
• `` -&gt; `...2`</code></pre>
</div>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="co"># gi &lt;- ieugwasr::gwasinfo(traits$opengwasid)</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a><span class="co"># saveRDS(gi, here("data", "gi.rds"))</span></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>gi <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"gi.rds"</span>))</span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"all.rdata"</span>))</span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>coloc_res <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"cis_coloc_res.rds"</span>))</span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>coloc_res <span class="ot">&lt;-</span> coloc_res <span class="sc">%&gt;%</span></span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(id.outcome)) <span class="sc">%&gt;%</span></span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(., gi <span class="sc">%&gt;%</span> <span class="fu">select</span>(trait, <span class="at">id.outcome =</span> id), <span class="at">by=</span><span class="fu">c</span>(<span class="st">"outcome"</span><span class="ot">=</span><span class="st">"trait"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(outcome)) <span class="sc">%&gt;%</span></span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(., traits <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="at">outcome=</span>code, <span class="at">id.outcome=</span>opengwasid), <span class="at">by=</span><span class="st">"id.outcome"</span>)</span>
<span id="cb23-11"><a href="#cb23-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-12"><a href="#cb23-12" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> res <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(outcome)) <span class="sc">%&gt;%</span></span>
<span id="cb23-13"><a href="#cb23-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(., traits <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="at">outcome=</span>code, <span class="at">id.outcome=</span>opengwasid), <span class="at">by=</span><span class="st">"id.outcome"</span>)</span>
<span id="cb23-14"><a href="#cb23-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-15"><a href="#cb23-15" aria-hidden="true" tabindex="-1"></a>res_cis <span class="ot">&lt;-</span> res_cis <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(outcome)) <span class="sc">%&gt;%</span></span>
<span id="cb23-16"><a href="#cb23-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(., traits <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="at">outcome=</span>code, <span class="at">id.outcome=</span>opengwasid), <span class="at">by=</span><span class="st">"id.outcome"</span>)</span>
<span id="cb23-17"><a href="#cb23-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-18"><a href="#cb23-18" aria-hidden="true" tabindex="-1"></a>res_trans <span class="ot">&lt;-</span> res_trans <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(outcome)) <span class="sc">%&gt;%</span></span>
<span id="cb23-19"><a href="#cb23-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(., traits <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="at">outcome=</span>code, <span class="at">id.outcome=</span>opengwasid), <span class="at">by=</span><span class="st">"id.outcome"</span>)</span>
<span id="cb23-20"><a href="#cb23-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-21"><a href="#cb23-21" aria-hidden="true" tabindex="-1"></a>res_trans_sf <span class="ot">&lt;-</span> res_trans_sf <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(outcome)) <span class="sc">%&gt;%</span></span>
<span id="cb23-22"><a href="#cb23-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(., traits <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="at">outcome=</span>code, <span class="at">id.outcome=</span>opengwasid), <span class="at">by=</span><span class="st">"id.outcome"</span>)</span>
<span id="cb23-23"><a href="#cb23-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-24"><a href="#cb23-24" aria-hidden="true" tabindex="-1"></a>res_trans_sf_relaxed <span class="ot">&lt;-</span> res_trans_sf_relaxed <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(outcome)) <span class="sc">%&gt;%</span></span>
<span id="cb23-25"><a href="#cb23-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(., traits <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="at">outcome=</span>code, <span class="at">id.outcome=</span>opengwasid), <span class="at">by=</span><span class="st">"id.outcome"</span>)</span>
<span id="cb23-26"><a href="#cb23-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-27"><a href="#cb23-27" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(coloc_res<span class="sc">$</span>outcome <span class="sc">%in%</span> traits<span class="sc">$</span>code)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
TRUE 
  96 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(coloc_res<span class="sc">$</span>outcome <span class="sc">%in%</span> res_cis<span class="sc">$</span>outcome)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
TRUE 
  96 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(coloc_res<span class="sc">$</span>outcome <span class="sc">%in%</span> res<span class="sc">$</span>outcome)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
TRUE 
  96 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(coloc_res<span class="sc">$</span>id.outcome <span class="sc">%in%</span> res<span class="sc">$</span>id.outcome)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
TRUE 
  96 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>coloc_res_s <span class="ot">&lt;-</span> <span class="fu">subset</span>(coloc_res, PP.H4.abf <span class="sc">&gt;=</span> <span class="fl">0.8</span>)</span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>res_cis_coloc <span class="ot">&lt;-</span> <span class="fu">subset</span>(res_cis, <span class="fu">paste</span>(id.exposure, id.outcome) <span class="sc">%in%</span> <span class="fu">paste</span>(coloc_res_s<span class="sc">$</span>id.exposure, coloc_res_s<span class="sc">$</span>id.outcome))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-hash="all_cis_mr_cache/html/unnamed-chunk-3_4b5b52dbb8b292ad1965c7f8c022c28b">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>cis_mr <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(</span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>    res_cis,</span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>    coloc_res <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(id.exposure, id.outcome, <span class="at">H4=</span>PP.H4.abf, <span class="at">H3=</span>PP.H3.abf, <span class="at">leadsnp=</span>snp),</span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">by=</span><span class="fu">c</span>(<span class="st">"id.exposure"</span>, <span class="st">"id.outcome"</span>)</span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">p.adjust</span>(pval, <span class="st">"fdr"</span>) <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span></span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a><span class="fu">arrange</span>(<span class="fu">desc</span>(H4)) <span class="sc">%&gt;%</span></span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a><span class="fu">select</span>(exposure, outcome, id.outcome, b, se, pval, H4, H3, leadsnp)</span>
<span id="cb32-8"><a href="#cb32-8" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="sc">!</span><span class="fu">is.na</span>(cis_mr<span class="sc">$</span>H4))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
TRUE 
  96 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(cis_mr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  exposure outcome       id.outcome           b         se         pval
1      CD5      MS         ieu-b-18 -1.00904713 0.12858303 4.246289e-15
2     CD58      MS         ieu-b-18  0.84188799 0.10325554 3.537107e-16
3     FGF5     CAD ebi-a-GCST003116  0.07408077 0.01516453 1.033584e-06
4    TGFB1     CAD ebi-a-GCST003116  0.26488749 0.05194817 3.413225e-07
5    EPHA1      AD          ieu-b-2 -0.41397484 0.07762028 9.642607e-08
6     IL7R      MS         ieu-b-18  0.12676867 0.02108730 1.836887e-09
         H4           H3     leadsnp
1 0.9996982 0.0003018188 11:60294184
2 0.9994966 0.0005034191 1:116592485
3 0.9982661 0.0003778683  4:80682779
4 0.9980567 0.0015842778 19:41325199
5 0.9975689 0.0022909378 7:142611217
6 0.9974652 0.0025302500  5:35374841</code></pre>
</div>
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(cis_mr, <span class="at">file=</span><span class="fu">here</span>(<span class="st">"results"</span>, <span class="st">"cis_mr.csv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Show MR results that have IVW / Wald ratio P val bonferroni corrected significant</p>
<div class="cell" data-hash="all_cis_mr_cache/html/unnamed-chunk-4_ff9a3e60b452b9798e058afe97c19c19">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>res <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(method) <span class="sc">%&gt;%</span></span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n=</span><span class="fu">n</span>(), <span class="at">nsig =</span> <span class="fu">sum</span>(fdr <span class="sc">&lt;</span> <span class="fl">0.05</span>, <span class="at">na.rm=</span>T))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 2 × 3
  method                        n  nsig
  &lt;chr&gt;                     &lt;int&gt; &lt;int&gt;
1 Inverse variance weighted 20275    78
2 Wald ratio                 2159    30</code></pre>
</div>
</div>
<div class="cell" data-hash="all_cis_mr_cache/html/unnamed-chunk-5_6247c405d69d43d76a9161e5c4166a1a">
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>ressig <span class="ot">&lt;-</span> <span class="fu">subset</span>(res, fdr <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(ressig)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 108  10</code></pre>
</div>
</div>
<p>Compare forward MR against reverse PRS results</p>
<div class="cell" data-hash="all_cis_mr_cache/html/unnamed-chunk-6_7f84570eb739a0b004dcddf876771624">
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a>temp <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(<span class="fu">subset</span>(res, method <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Inverse variance weighted"</span>, <span class="st">"Wald ratio"</span>)), prs_pairs, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"id.exposure"</span><span class="ot">=</span><span class="st">"prot"</span>, <span class="st">"id.outcome"</span><span class="ot">=</span><span class="st">"opengwasid"</span>))</span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>temp<span class="sc">$</span>mr_sig <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(temp<span class="sc">$</span>pval.x, <span class="st">"fdr"</span>) <span class="sc">&lt;</span> <span class="fl">0.05</span></span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(temp, <span class="fu">aes</span>(beta, b)) <span class="sc">+</span></span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">0</span>, <span class="at">linetype=</span><span class="st">"dotted"</span>) <span class="sc">+</span></span>
<span id="cb41-5"><a href="#cb41-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">0</span>, <span class="at">linetype=</span><span class="st">"dotted"</span>) <span class="sc">+</span></span>
<span id="cb41-6"><a href="#cb41-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">colour=</span>mr_sig)) <span class="sc">+</span></span>
<span id="cb41-7"><a href="#cb41-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method=</span><span class="st">"lm"</span>) <span class="sc">+</span></span>
<span id="cb41-8"><a href="#cb41-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">"PRS disease -&gt; protein"</span>, <span class="at">y=</span><span class="st">"MR protein -&gt; disease"</span>) <span class="sc">+</span></span>
<span id="cb41-9"><a href="#cb41-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(mr_sig <span class="sc">~</span> .)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`geom_smooth()` using formula = 'y ~ x'</code></pre>
</div>
<div class="cell-output-display">
<p><img src="all_cis_mr_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Cis vs trans</p>
<div class="cell" data-hash="all_cis_mr_cache/html/unnamed-chunk-7_4fb6e45036da5a763e499cbc511f5162">
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">p.adjust</span>(res_cis_coloc<span class="sc">$</span>p, <span class="st">"fdr"</span>) <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
TRUE 
  45 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">p.adjust</span>(res_trans<span class="sc">$</span>p, <span class="st">"fdr"</span>) <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
FALSE  TRUE 
21279   186 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="fu">inner_join</span>(res_cis_coloc, res_trans, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"id.exposure"</span>, <span class="st">"id.outcome"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a>{<span class="fu">table</span>(.<span class="sc">$</span>pval.x <span class="sc">&lt;</span> <span class="fl">0.05</span>, .<span class="sc">$</span>pval.y <span class="sc">&lt;</span> <span class="fl">0.05</span>)}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>      
       FALSE TRUE
  TRUE    34    8</code></pre>
</div>
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(<span class="fu">lm</span>(<span class="fu">I</span>(pval.x <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">~</span> <span class="fu">I</span>(pval.y <span class="sc">&lt;</span> <span class="fl">0.05</span>), <span class="at">data=</span><span class="fu">inner_join</span>(res_cis_coloc, res_trans, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"id.exposure"</span>, <span class="st">"id.outcome"</span>))))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in summary.lm(lm(I(pval.x &lt; 0.05) ~ I(pval.y &lt; 0.05), data =
inner_join(res_cis_coloc, : essentially perfect fit: summary may be unreliable</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = I(pval.x &lt; 0.05) ~ I(pval.y &lt; 0.05), data = inner_join(res_cis_coloc, 
    res_trans, by = c("id.exposure", "id.outcome")))

Residuals:
       Min         1Q     Median         3Q        Max 
-2.122e-15  0.000e+00  0.000e+00  0.000e+00  3.031e-16 

Coefficients:
                       Estimate Std. Error    t value Pr(&gt;|t|)    
(Intercept)           1.000e+00  6.151e-17  1.626e+16   &lt;2e-16 ***
I(pval.y &lt; 0.05)TRUE -3.031e-16  1.409e-16 -2.151e+00   0.0376 *  
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 3.587e-16 on 40 degrees of freedom
Multiple R-squared:  0.5258,    Adjusted R-squared:  0.514 
F-statistic: 44.36 on 1 and 40 DF,  p-value: 5.589e-08</code></pre>
</div>
</div>
<p>Are cis and trans MR results occurring more than expected by chance? For cis and trans MRs across protein pairs identify how many have p-values &lt; alpha, then scramble cis and trans protein pairs 10000 times and determine if the observed number is larger than the distribution of null counts.</p>
<div class="cell" data-hash="all_cis_mr_cache/html/unnamed-chunk-8_e67763bd33beeca7a0410afe733da15a">
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a>nperm <span class="ot">&lt;-</span> <span class="dv">10000</span></span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a>perm_res <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="fu">c</span>(<span class="fl">0.05</span>, <span class="fl">0.01</span>, <span class="fl">0.001</span>, <span class="fl">0.05</span><span class="sc">/</span><span class="fu">nrow</span>(temp)), <span class="cf">function</span>(thresh){</span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a>    temp <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(res_cis, res_trans, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"id.exposure"</span>, <span class="st">"id.outcome"</span>))</span>
<span id="cb52-4"><a href="#cb52-4" aria-hidden="true" tabindex="-1"></a>    ncis <span class="ot">&lt;-</span> <span class="fu">sum</span>(temp<span class="sc">$</span>pval.x <span class="sc">&lt;</span> thresh)</span>
<span id="cb52-5"><a href="#cb52-5" aria-hidden="true" tabindex="-1"></a>    ntrans <span class="ot">&lt;-</span> <span class="fu">sum</span>(temp<span class="sc">$</span>pval.y <span class="sc">&lt;</span> thresh, <span class="at">na.rm=</span>T)</span>
<span id="cb52-6"><a href="#cb52-6" aria-hidden="true" tabindex="-1"></a>    perms <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="dv">1</span><span class="sc">:</span>nperm, <span class="cf">function</span>(x) <span class="fu">sum</span>(<span class="fu">sample</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(temp), ncis, <span class="at">replace=</span><span class="cn">FALSE</span>) <span class="sc">%in%</span> <span class="fu">sample</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(temp), ntrans, <span class="at">replace=</span><span class="cn">FALSE</span>))) <span class="sc">%&gt;%</span> <span class="fu">sort</span>()</span>
<span id="cb52-7"><a href="#cb52-7" aria-hidden="true" tabindex="-1"></a>    real <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(res_cis, res_trans, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"id.exposure"</span>, <span class="st">"id.outcome"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb52-8"><a href="#cb52-8" aria-hidden="true" tabindex="-1"></a>    {<span class="fu">sum</span>(.<span class="sc">$</span>pval.x <span class="sc">&lt;</span> thresh <span class="sc">&amp;</span> .<span class="sc">$</span>pval.y <span class="sc">&lt;</span> thresh, <span class="at">na.rm=</span>T)}</span>
<span id="cb52-9"><a href="#cb52-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">tibble</span>(<span class="at">thresh=</span>thresh, <span class="at">nsig=</span>real, <span class="at">maxperm=</span><span class="fu">max</span>(perms), <span class="at">medperm=</span><span class="fu">median</span>(perms), <span class="at">p_perm=</span><span class="fu">sum</span>(real <span class="sc">&lt;</span> perms)<span class="sc">/</span>nperm)</span>
<span id="cb52-10"><a href="#cb52-10" aria-hidden="true" tabindex="-1"></a>}) <span class="sc">%&gt;%</span> <span class="fu">bind_rows</span>()</span>
<span id="cb52-11"><a href="#cb52-11" aria-hidden="true" tabindex="-1"></a>perm_res</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 4 × 5
     thresh  nsig maxperm medperm p_perm
      &lt;dbl&gt; &lt;int&gt;   &lt;int&gt;   &lt;dbl&gt;  &lt;dbl&gt;
1 0.05        120     143     107 0.0807
2 0.01         20      31      14 0.0614
3 0.001         2       8       1 0.191 
4 0.0000327     1       4       0 0.0207</code></pre>
</div>
</div>
<p>Cis and trans MR results co-occur more than expected by chance.</p>
<p>Do effect sizes correspond?</p>
<div class="cell" data-hash="all_cis_mr_cache/html/unnamed-chunk-9_446bbc66155e7893318c5744b4b5d9ff">
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="fu">inner_join</span>(res_cis, res_trans, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"id.exposure"</span>, <span class="st">"id.outcome"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(pval.x <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span></span>
<span id="cb54-3"><a href="#cb54-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(., <span class="fu">aes</span>(<span class="at">x=</span>b.x, <span class="at">y=</span>b.y)) <span class="sc">+</span></span>
<span id="cb54-4"><a href="#cb54-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb54-5"><a href="#cb54-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_smooth</span>(<span class="at">method=</span><span class="st">"lm"</span>) <span class="sc">+</span> </span>
<span id="cb54-6"><a href="#cb54-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x=</span><span class="st">"cis-MR estimate"</span>, <span class="at">y=</span><span class="st">"trans-MR estimate"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`geom_smooth()` using formula = 'y ~ x'</code></pre>
</div>
<div class="cell-output-display">
<p><img src="all_cis_mr_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>No concordance between cis and trans - just like in GoDMC! Restrict to cis coloc</p>
<div class="cell" data-hash="all_cis_mr_cache/html/unnamed-chunk-10_6ff12f072e95e3bf4287107d71532868">
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="fu">inner_join</span>(res_cis_coloc, res_trans, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"id.exposure"</span>, <span class="st">"id.outcome"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(pval.x <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span></span>
<span id="cb56-3"><a href="#cb56-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(., <span class="fu">aes</span>(<span class="at">x=</span>b.x, <span class="at">y=</span>b.y)) <span class="sc">+</span></span>
<span id="cb56-4"><a href="#cb56-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb56-5"><a href="#cb56-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_smooth</span>(<span class="at">method=</span><span class="st">"lm"</span>) <span class="sc">+</span> </span>
<span id="cb56-6"><a href="#cb56-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x=</span><span class="st">"cis-MR estimate"</span>, <span class="at">y=</span><span class="st">"trans-MR estimate"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`geom_smooth()` using formula = 'y ~ x'</code></pre>
</div>
<div class="cell-output-display">
<p><img src="all_cis_mr_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell" data-hash="all_cis_mr_cache/html/unnamed-chunk-11_325ca50446791a428d275ae644d894b9">
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="fu">inner_join</span>(res_cis_coloc, res_trans, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"id.exposure"</span>, <span class="st">"id.outcome"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(pval.x <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span></span>
<span id="cb58-3"><a href="#cb58-3" aria-hidden="true" tabindex="-1"></a>    {<span class="fu">summary</span>(<span class="fu">lm</span>(b.y <span class="sc">~</span> b.x, .))}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = b.y ~ b.x, data = .)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.52337 -0.15126 -0.04964  0.12570  1.08342 

Coefficients:
             Estimate Std. Error t value Pr(&gt;|t|)  
(Intercept)  0.086206   0.045513   1.894   0.0655 .
b.x         -0.003395   0.099608  -0.034   0.9730  
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 0.2909 on 40 degrees of freedom
Multiple R-squared:  2.905e-05, Adjusted R-squared:  -0.02497 
F-statistic: 0.001162 on 1 and 40 DF,  p-value: 0.973</code></pre>
</div>
</div>
<div class="cell" data-hash="all_cis_mr_cache/html/unnamed-chunk-12_ffbbb639eae1b643088e47f607630e32">
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a><span class="fu">inner_join</span>(res_cis, res_trans, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"id.exposure"</span>, <span class="st">"id.outcome"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a>    {<span class="fu">summary</span>(<span class="fu">lm</span>(b.y <span class="sc">~</span> b.x, .))}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = b.y ~ b.x, data = .)

Residuals:
    Min      1Q  Median      3Q     Max 
-8.9733 -0.0784  0.0049  0.0863 13.1980 

Coefficients:
             Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept) -0.009173   0.002327  -3.941 8.14e-05 ***
b.x          0.046808   0.016161   2.896  0.00378 ** 
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 0.278 on 14276 degrees of freedom
Multiple R-squared:  0.0005873, Adjusted R-squared:  0.0005173 
F-statistic: 8.389 on 1 and 14276 DF,  p-value: 0.003781</code></pre>
</div>
</div>
<div class="cell" data-hash="all_cis_mr_cache/html/unnamed-chunk-13_f34276cb1cf16e0676fd00671089e027">
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="fu">inner_join</span>(res_cis, res_trans, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"id.exposure"</span>, <span class="st">"id.outcome"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb62-2"><a href="#cb62-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(pval.x <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span></span>
<span id="cb62-3"><a href="#cb62-3" aria-hidden="true" tabindex="-1"></a>    {<span class="fu">summary</span>(<span class="fu">lm</span>(b.y <span class="sc">~</span> b.x, .))}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = b.y ~ b.x, data = .)

Residuals:
    Min      1Q  Median      3Q     Max 
-8.9482 -0.0789  0.0044  0.0849  6.5161 

Coefficients:
             Estimate Std. Error t value Pr(&gt;|t|)   
(Intercept) -0.003015   0.012169  -0.248  0.80439   
b.x          0.124833   0.039318   3.175  0.00154 **
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 0.4206 on 1197 degrees of freedom
Multiple R-squared:  0.008351,  Adjusted R-squared:  0.007523 
F-statistic: 10.08 on 1 and 1197 DF,  p-value: 0.001537</code></pre>
</div>
</div>
<section id="forward-vs-reverse-mr" class="level2">
<h2 class="anchored" data-anchor-id="forward-vs-reverse-mr">Forward vs reverse MR</h2>
<p>Do forward and reverse MR results agree?</p>
<div class="cell" data-hash="all_cis_mr_cache/html/unnamed-chunk-14_a0e4d401121a1c9fe431ce51d32909d1">
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="fu">bind_rows</span>(res_cis <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">what=</span><span class="st">"cis"</span>), res_trans <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">what=</span><span class="st">"trans"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb64-2"><a href="#cb64-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(pval)) <span class="sc">%&gt;%</span></span>
<span id="cb64-3"><a href="#cb64-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">inner_join</span>(., prs_pairs, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"id.exposure"</span><span class="ot">=</span><span class="st">"prot"</span>, <span class="st">"id.outcome"</span><span class="ot">=</span><span class="st">"opengwasid"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb64-4"><a href="#cb64-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">mr_sig =</span> <span class="fu">p.adjust</span>(pval.x, <span class="st">"fdr"</span>) <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span></span>
<span id="cb64-5"><a href="#cb64-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(., <span class="fu">aes</span>(beta, b)) <span class="sc">+</span></span>
<span id="cb64-6"><a href="#cb64-6" aria-hidden="true" tabindex="-1"></a>        <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">0</span>, <span class="at">linetype=</span><span class="st">"dotted"</span>) <span class="sc">+</span></span>
<span id="cb64-7"><a href="#cb64-7" aria-hidden="true" tabindex="-1"></a>        <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">0</span>, <span class="at">linetype=</span><span class="st">"dotted"</span>) <span class="sc">+</span></span>
<span id="cb64-8"><a href="#cb64-8" aria-hidden="true" tabindex="-1"></a>        <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">colour=</span>mr_sig)) <span class="sc">+</span></span>
<span id="cb64-9"><a href="#cb64-9" aria-hidden="true" tabindex="-1"></a>        <span class="fu">geom_smooth</span>(<span class="at">method=</span><span class="st">"lm"</span>) <span class="sc">+</span></span>
<span id="cb64-10"><a href="#cb64-10" aria-hidden="true" tabindex="-1"></a>        <span class="fu">labs</span>(<span class="at">x=</span><span class="st">"PRS disease -&gt; protein"</span>, <span class="at">y=</span><span class="st">"MR protein -&gt; disease"</span>) <span class="sc">+</span></span>
<span id="cb64-11"><a href="#cb64-11" aria-hidden="true" tabindex="-1"></a>        <span class="fu">facet_grid</span>(mr_sig <span class="sc">~</span> what)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`geom_smooth()` using formula = 'y ~ x'</code></pre>
</div>
<div class="cell-output-display">
<p><img src="all_cis_mr_files/figure-html/unnamed-chunk-14-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Use steiger filtering. Note only trans instruments had any evidence of being reverse causal.</p>
<div class="cell" data-hash="all_cis_mr_cache/html/unnamed-chunk-15_f17a90bd6b1d69709e5416a0520c8dc9">
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(dat<span class="sc">$</span>rsq.exposure <span class="sc">&gt;</span> dat<span class="sc">$</span>rsq.outcome, dat<span class="sc">$</span>cistrans.exposure)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>       
           cis  trans
  FALSE      0    231
  TRUE   15426 122946</code></pre>
</div>
<div class="sourceCode cell-code" id="cb68"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(dat<span class="sc">$</span>steiger_pval <span class="sc">&lt;</span> <span class="fl">0.05</span>, dat<span class="sc">$</span>cistrans.exposure)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>       
           cis  trans
  FALSE     25   2569
  TRUE   15401 120608</code></pre>
</div>
</div>
<p>All reverse instruments are trans!</p>
<div class="cell" data-hash="all_cis_mr_cache/html/unnamed-chunk-16_f2c06114156bdc055f6576c498c92537">
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a><span class="fu">bind_rows</span>(</span>
<span id="cb70-2"><a href="#cb70-2" aria-hidden="true" tabindex="-1"></a>    res_cis <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">what=</span><span class="st">"cis"</span>), </span>
<span id="cb70-3"><a href="#cb70-3" aria-hidden="true" tabindex="-1"></a>    res_cis_coloc <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">what=</span><span class="st">"cis+coloc"</span>), </span>
<span id="cb70-4"><a href="#cb70-4" aria-hidden="true" tabindex="-1"></a>    res_trans <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">what=</span><span class="st">"trans"</span>), </span>
<span id="cb70-5"><a href="#cb70-5" aria-hidden="true" tabindex="-1"></a>    res_trans_sf <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">what=</span><span class="st">"trans sf"</span>),</span>
<span id="cb70-6"><a href="#cb70-6" aria-hidden="true" tabindex="-1"></a>    res_trans_sf_relaxed <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">what=</span><span class="st">"trans sf strict"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb70-7"><a href="#cb70-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(pval), <span class="fu">abs</span>(b) <span class="sc">&lt;</span> <span class="dv">5</span>) <span class="sc">%&gt;%</span></span>
<span id="cb70-8"><a href="#cb70-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">inner_join</span>(., prs_pairs, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"id.exposure"</span><span class="ot">=</span><span class="st">"prot"</span>, <span class="st">"id.outcome"</span><span class="ot">=</span><span class="st">"opengwasid"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb70-9"><a href="#cb70-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">mr_sig =</span> <span class="fu">p.adjust</span>(pval.x, <span class="st">"fdr"</span>) <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span></span>
<span id="cb70-10"><a href="#cb70-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(., <span class="fu">aes</span>(beta, b)) <span class="sc">+</span></span>
<span id="cb70-11"><a href="#cb70-11" aria-hidden="true" tabindex="-1"></a>        <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">0</span>, <span class="at">linetype=</span><span class="st">"dotted"</span>) <span class="sc">+</span></span>
<span id="cb70-12"><a href="#cb70-12" aria-hidden="true" tabindex="-1"></a>        <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">0</span>, <span class="at">linetype=</span><span class="st">"dotted"</span>) <span class="sc">+</span></span>
<span id="cb70-13"><a href="#cb70-13" aria-hidden="true" tabindex="-1"></a>        <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">colour=</span>mr_sig)) <span class="sc">+</span></span>
<span id="cb70-14"><a href="#cb70-14" aria-hidden="true" tabindex="-1"></a>        <span class="fu">geom_smooth</span>(<span class="at">method=</span><span class="st">"lm"</span>) <span class="sc">+</span></span>
<span id="cb70-15"><a href="#cb70-15" aria-hidden="true" tabindex="-1"></a>        <span class="fu">labs</span>(<span class="at">x=</span><span class="st">"PRS disease -&gt; protein"</span>, <span class="at">y=</span><span class="st">"MR protein -&gt; disease"</span>) <span class="sc">+</span></span>
<span id="cb70-16"><a href="#cb70-16" aria-hidden="true" tabindex="-1"></a>        <span class="fu">facet_grid</span>(mr_sig <span class="sc">~</span> what)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`geom_smooth()` using formula = 'y ~ x'</code></pre>
</div>
<div class="cell-output-display">
<p><img src="all_cis_mr_files/figure-html/unnamed-chunk-16-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="prs-discovers-forward-mr" class="level2">
<h2 class="anchored" data-anchor-id="prs-discovers-forward-mr">PRS discovers forward MR?</h2>
<div class="cell" data-hash="all_cis_mr_cache/html/unnamed-chunk-17_b7e218e0967cf3321cf279758b0ef452">
<div class="sourceCode cell-code" id="cb72"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a>cis_prs <span class="ot">&lt;-</span> <span class="fu">subset</span>(res_cis, <span class="fu">paste</span>(id.exposure, id.outcome) <span class="sc">%in%</span> <span class="fu">paste</span>(prs_pairs<span class="sc">$</span>prot, prs_pairs<span class="sc">$</span>opengwasid))</span>
<span id="cb72-2"><a href="#cb72-2" aria-hidden="true" tabindex="-1"></a>cis_prs<span class="sc">$</span>fdr <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(cis_prs<span class="sc">$</span>pval, <span class="st">"fdr"</span>)</span>
<span id="cb72-3"><a href="#cb72-3" aria-hidden="true" tabindex="-1"></a>a <span class="ot">&lt;-</span> <span class="fu">subset</span>(cis_prs, fdr <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">code =</span> <span class="fu">paste</span>(id.exposure, id.outcome))</span>
<span id="cb72-4"><a href="#cb72-4" aria-hidden="true" tabindex="-1"></a>res_cis<span class="sc">$</span>fdr <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(res_cis<span class="sc">$</span>pval, <span class="st">"fdr"</span>)</span>
<span id="cb72-5"><a href="#cb72-5" aria-hidden="true" tabindex="-1"></a>b <span class="ot">&lt;-</span> <span class="fu">subset</span>(res_cis, fdr <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">code =</span> <span class="fu">paste</span>(id.exposure, id.outcome))</span>
<span id="cb72-6"><a href="#cb72-6" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(a<span class="sc">$</span>code <span class="sc">%in%</span> b<span class="sc">$</span>code)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 22</code></pre>
</div>
<div class="sourceCode cell-code" id="cb74"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="fu">tribble</span>(<span class="sc">~</span>what, <span class="sc">~</span>count,</span>
<span id="cb74-2"><a href="#cb74-2" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Total tested - forward MR"</span>, <span class="fu">nrow</span>(res_cis),</span>
<span id="cb74-3"><a href="#cb74-3" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Total discovered - forward MR"</span>, <span class="fu">sum</span>(res_cis<span class="sc">$</span>fdr <span class="sc">&lt;</span> <span class="fl">0.05</span>),</span>
<span id="cb74-4"><a href="#cb74-4" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Total tested - reverse MR filter"</span>, <span class="fu">nrow</span>(cis_prs),</span>
<span id="cb74-5"><a href="#cb74-5" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Total discovered - reverse MR filter"</span>, <span class="fu">sum</span>(cis_prs<span class="sc">$</span>fdr <span class="sc">&lt;</span> <span class="fl">0.05</span>),</span>
<span id="cb74-6"><a href="#cb74-6" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Discovery overlap"</span>, <span class="fu">sum</span>(a<span class="sc">$</span>code <span class="sc">%in%</span> b<span class="sc">$</span>code)</span>
<span id="cb74-7"><a href="#cb74-7" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 5 × 2
  what                                 count
  &lt;chr&gt;                                &lt;int&gt;
1 Total tested - forward MR            15246
2 Total discovered - forward MR           94
3 Total tested - reverse MR filter       980
4 Total discovered - reverse MR filter    30
5 Discovery overlap                       22</code></pre>
</div>
</div>
<p>Cis MR is substantially better powered than PRS for identifying forward MR assocs. But 8.5% extra forward MR assocs discovered that wouldn’t have been found by cis MR.</p>
</section>
<section id="bidirectional-effects" class="level2">
<h2 class="anchored" data-anchor-id="bidirectional-effects">Bidirectional effects?</h2>
<p>Using cis for forward and any with !steiger for reverse</p>
<div class="cell" data-hash="all_cis_mr_cache/html/unnamed-chunk-18_d35ba088844ac5ec284b651782f24a78">
<div class="sourceCode cell-code" id="cb76"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a>rev_ids <span class="ot">&lt;-</span> <span class="fu">paste</span>(dat<span class="sc">$</span>id.exposure[<span class="sc">!</span>dat<span class="sc">$</span>steiger_dir], dat<span class="sc">$</span>id.outcome[<span class="sc">!</span>dat<span class="sc">$</span>steiger_dir])</span>
<span id="cb76-2"><a href="#cb76-2" aria-hidden="true" tabindex="-1"></a>cis_mr_ids <span class="ot">&lt;-</span> <span class="fu">paste</span>(res_cis_coloc<span class="sc">$</span>id.exposure, res_cis_coloc<span class="sc">$</span>id.outcome)</span>
<span id="cb76-3"><a href="#cb76-3" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(rev_ids <span class="sc">%in%</span> cis_mr_ids)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
FALSE 
  231 </code></pre>
</div>
</div>
<div class="cell" data-hash="all_cis_mr_cache/html/unnamed-chunk-19_479f1631e03da0fd8c7984818e8530d0">
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a>bid <span class="ot">&lt;-</span> <span class="fu">subset</span>(res_cis_coloc, cis_mr_ids <span class="sc">%in%</span> rev_ids)</span>
<span id="cb78-2"><a href="#cb78-2" aria-hidden="true" tabindex="-1"></a>bid</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] id.exposure id.outcome  exposure    method      nsnp        b          
[7] se          pval        outcome    
&lt;0 rows&gt; (or 0-length row.names)</code></pre>
</div>
</div>
<div class="cell" data-hash="all_cis_mr_cache/html/unnamed-chunk-20_4321df57a8d2f46f97f8748276be7ae1">
<div class="sourceCode cell-code" id="cb80"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a>rev_ids_prs <span class="ot">&lt;-</span> <span class="fu">paste</span>(prs_pairs<span class="sc">$</span>prot, prs_pairs<span class="sc">$</span>opengwasid)</span>
<span id="cb80-2"><a href="#cb80-2" aria-hidden="true" tabindex="-1"></a>cis_mr_ids <span class="ot">&lt;-</span> <span class="fu">paste</span>(res_cis_coloc<span class="sc">$</span>id.exposure, res_cis_coloc<span class="sc">$</span>id.outcome)</span>
<span id="cb80-3"><a href="#cb80-3" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(rev_ids_prs <span class="sc">%in%</span> cis_mr_ids)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
FALSE  TRUE 
 2564    12 </code></pre>
</div>
</div>
<div class="cell" data-hash="all_cis_mr_cache/html/unnamed-chunk-21_e8916720c070a5fb1aa1f6e867ee02d8">
<div class="sourceCode cell-code" id="cb82"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a>bidprs <span class="ot">&lt;-</span> <span class="fu">subset</span>(res_cis_coloc, cis_mr_ids <span class="sc">%in%</span> rev_ids_prs)</span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a>bidprs</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>      id.exposure       id.outcome exposure     method nsnp           b
1308       BTN2A1       ieu-a-1126   BTN2A1 Wald ratio    1 -0.15303021
1313       BTN2A1         ieu-b-18   BTN2A1 Wald ratio    1  0.25949643
1315       BTN2A1         ieu-b-42   BTN2A1 Wald ratio    1 -0.45843999
2537         CD33          ieu-b-2     CD33 Wald ratio    1  0.04904013
2648          CD5         ieu-b-18      CD5 Wald ratio    1 -1.00904713
6245        FURIN ebi-a-GCST003116    FURIN Wald ratio    1  0.27589234
7906         IL6R ebi-a-GCST003116     IL6R Wald ratio    1 -0.04496194
7961         IL7R         ieu-b-18     IL7R Wald ratio    1  0.12676867
8362         KLK1        ieu-a-984     KLK1 Wald ratio    1  0.08564168
8939        LRIG1         ieu-a-26    LRIG1 Wald ratio    1  0.10069053
10771       PCSK9 ebi-a-GCST003116    PCSK9 Wald ratio    1  0.22459591
14782        UMOD       ieu-a-1102     UMOD Wald ratio    1  0.22882153
               se         pval             outcome
1308  0.025469364 1.873592e-09       Breast cancer
1313  0.060216211 1.636895e-05                  MS
1315  0.043875795 1.487270e-25       Schizophrenia
2537  0.012559057 9.431820e-05                  AD
2648  0.128583034 4.246289e-15                  MS
6245  0.055415267 6.403459e-07                 CAD
7906  0.008679061 2.212830e-07                 CAD
7961  0.021087304 1.836887e-09                  MS
8362  0.018298109 2.863762e-06 Lung adenocarcinoma
8939  0.024738306 4.696730e-05                 T2D
10771 0.050136420 7.474953e-06                 CAD
14782 0.021842055 1.111305e-25                 CKD</code></pre>
</div>
</div>
<p>Some of these PRS results could just be driven by strong cis signals e.g.&nbsp;PCSK9.</p>
<div class="cell" data-hash="all_cis_mr_cache/html/unnamed-chunk-22_b8f95f1fda18b6a7edbe6dded84366af">
<div class="sourceCode cell-code" id="cb84"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a>bidprs<span class="sc">$</span>ids <span class="ot">&lt;-</span> <span class="fu">paste</span>(bidprs<span class="sc">$</span>id.exposure, bidprs<span class="sc">$</span>id.outcome)</span>
<span id="cb84-2"><a href="#cb84-2" aria-hidden="true" tabindex="-1"></a>bid_trans <span class="ot">&lt;-</span> <span class="fu">subset</span>(res_trans, <span class="fu">paste</span>(id.exposure, id.outcome) <span class="sc">%in%</span> bidprs<span class="sc">$</span>ids) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">fdr =</span> <span class="fu">p.adjust</span>(pval, <span class="st">"fdr"</span>))</span>
<span id="cb84-3"><a href="#cb84-3" aria-hidden="true" tabindex="-1"></a>bid_trans <span class="sc">%&gt;%</span> <span class="fu">subset</span>(fdr <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] id.exposure id.outcome  exposure    method      nsnp        b          
 [7] se          pval        outcome     fdr        
&lt;0 rows&gt; (or 0-length row.names)</code></pre>
</div>
</div>
<p><strong>Are these plausible bi-directional associations?</strong></p>
</section>
<section id="perform-instrument-heterogeneity-analysis" class="level2">
<h2 class="anchored" data-anchor-id="perform-instrument-heterogeneity-analysis">Perform instrument heterogeneity analysis</h2>
<div class="cell" data-hash="all_cis_mr_cache/html/unnamed-chunk-23_f7bb96bfec086e9c444db2b91392dc65">
<div class="sourceCode cell-code" id="cb86"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a>het2 <span class="ot">&lt;-</span> dat <span class="sc">%&gt;%</span></span>
<span id="cb86-2"><a href="#cb86-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">subset</span>(., <span class="fu">paste</span>(id.exposure, id.outcome) <span class="sc">%in%</span> <span class="fu">paste</span>(ressig<span class="sc">$</span>id.exposure, ressig<span class="sc">$</span>id.outcome)) <span class="sc">%&gt;%</span></span>
<span id="cb86-3"><a href="#cb86-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mr_heterogeneity</span>(<span class="at">method_list=</span><span class="st">"mr_ivw"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Not enough SNPs available for Heterogeneity analysis of 'AIF1' on 'ieu-b-18'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Not enough SNPs available for Heterogeneity analysis of 'AIF1' on 'ieu-b-42'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Not enough SNPs available for Heterogeneity analysis of 'ATP6AP2' on 'ieu-b-42'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Not enough SNPs available for Heterogeneity analysis of 'ATP6V1D' on 'ieu-b-18'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Not enough SNPs available for Heterogeneity analysis of 'DCN' on 'ieu-b-7'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Not enough SNPs available for Heterogeneity analysis of 'DCTN6' on 'ieu-b-85'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Not enough SNPs available for Heterogeneity analysis of 'DNPH1' on 'ieu-a-1126'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Not enough SNPs available for Heterogeneity analysis of 'DNPH1' on 'ieu-a-1127'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Not enough SNPs available for Heterogeneity analysis of 'EPHB6' on 'ieu-a-1102'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Not enough SNPs available for Heterogeneity analysis of 'FABP4' on 'ieu-a-26'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Not enough SNPs available for Heterogeneity analysis of 'FAM3C' on 'ieu-a-1102'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Not enough SNPs available for Heterogeneity analysis of 'FOXO3' on 'ieu-a-989'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Not enough SNPs available for Heterogeneity analysis of 'FOXO3' on 'ieu-b-18'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Not enough SNPs available for Heterogeneity analysis of 'FOXO3' on 'ieu-b-89'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Not enough SNPs available for Heterogeneity analysis of 'KLK10' on 'ieu-a-989'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Not enough SNPs available for Heterogeneity analysis of 'KLK11' on 'ieu-a-1102'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Not enough SNPs available for Heterogeneity analysis of 'KLK6' on 'ieu-a-1102'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Not enough SNPs available for Heterogeneity analysis of 'LEP' on 'ieu-a-1126'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Not enough SNPs available for Heterogeneity analysis of 'LEP' on 'ieu-a-1127'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Not enough SNPs available for Heterogeneity analysis of 'LEP' on 'ieu-a-26'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Not enough SNPs available for Heterogeneity analysis of 'POLR2F' on 'ebi-a-GCST003116'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Not enough SNPs available for Heterogeneity analysis of 'POLR2F' on 'ebi-a-GCST005838'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Not enough SNPs available for Heterogeneity analysis of 'POLR2F' on 'ebi-a-GCST006464'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Not enough SNPs available for Heterogeneity analysis of 'POLR2F' on 'ieu-b-18'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Not enough SNPs available for Heterogeneity analysis of 'PVR' on 'ieu-b-18'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Not enough SNPs available for Heterogeneity analysis of 'PVR' on 'ieu-b-2'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Not enough SNPs available for Heterogeneity analysis of 'SIGLEC5' on 'ieu-b-18'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Not enough SNPs available for Heterogeneity analysis of 'TFF3' on 'ieu-a-1102'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Not enough SNPs available for Heterogeneity analysis of 'TNFRSF10A' on 'ieu-b-18'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Not enough SNPs available for Heterogeneity analysis of 'XG' on 'ieu-a-26'</code></pre>
</div>
<div class="sourceCode cell-code" id="cb117"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb117-1"><a href="#cb117-1" aria-hidden="true" tabindex="-1"></a>het2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>   id.exposure       id.outcome
1       ADAM22         ieu-b-42
2          AMN         ieu-b-85
3         ART3 ebi-a-GCST005647
4       BTN2A1         ieu-b-42
5       BTN3A2         ieu-b-42
6         CA11 ebi-a-GCST003116
7        CD160         ieu-b-85
8        CD207 ebi-a-GCST003116
9         CD33          ieu-b-2
10        CD34       ieu-a-1120
11        CD34       ieu-a-1121
12        CD34       ieu-a-1126
13        CD34       ieu-a-1228
14        CD40         ieu-b-42
15        CDH1         ieu-b-18
16       CDH15         ieu-b-18
17        CDH2         ieu-a-26
18        CDSN         ieu-b-18
19      CGREF1         ieu-b-42
20       CRIP2          ieu-b-7
21       CXADR         ieu-b-42
22       DKKL1         ieu-b-18
23       EDIL3       ieu-a-1126
24       EPHB4       ieu-a-1120
25       EPHB4       ieu-a-1121
26       EPHB4       ieu-a-1228
27       ERBB4        ieu-a-984
28       ERBIN       ieu-a-1127
29          F3         ieu-b-18
30      FCGR2B          ieu-b-7
31       FGF21       ieu-a-1126
32       FURIN       ieu-a-1127
33         GRN ebi-a-GCST003116
34      IFNGR1         ieu-b-89
35         IL6 ebi-a-GCST003116
36      JCHAIN          ieu-b-2
37     KIR2DL3         ieu-b-18
38     KIR2DL3         ieu-b-42
39        LAYN       ieu-a-1126
40      LGALS4 ebi-a-GCST006464
41        LIFR       ieu-a-1120
42        LIFR       ieu-a-1228
43       MED18       ieu-a-1126
44       MED18       ieu-a-1127
45        MSMB         ieu-b-85
46        MZB1          ieu-b-2
47      NOTCH1         ieu-b-89
48       NTRK2         ieu-b-18
49      PCDH17       ieu-a-1126
50      PECAM1       ieu-a-1120
51      PECAM1       ieu-a-1121
52      PECAM1       ieu-a-1228
53       PILRA          ieu-b-2
54      PLA2G7 ebi-a-GCST003116
55        PLAT         ieu-b-18
56      PLXNB2         ieu-b-18
57       PTPRM       ieu-a-1120
58       PTPRM       ieu-a-1121
59       PTPRM       ieu-a-1126
60       PTPRM       ieu-a-1228
61        SCGN         ieu-b-18
62        SELE       ieu-a-1121
63        SELE       ieu-a-1126
64      SEMA4C       ieu-a-1120
65      SEMA4C       ieu-a-1121
66      SEMA4C       ieu-a-1126
67      SEMA4C       ieu-a-1127
68      SEMA4C       ieu-a-1228
69     SLITRK6         ieu-a-26
70       TGFB1 ebi-a-GCST003116
71    TNFRSF21       ieu-a-1102
72    TNFRSF6B         ieu-b-85
73       TRAF2 ebi-a-GCST005838
74       TREM2          ieu-b-2
75      TRIM21         ieu-b-89
76        UMOD       ieu-a-1102
77       VSIG4       ieu-a-1126
78       VSIG4       ieu-a-1127
                                                                              outcome
1                                                        schizophrenia || id:ieu-b-42
2                                                      Prostate cancer || id:ieu-b-85
3                                Amyotrophic lateral sclerosis || id:ebi-a-GCST005647
4                                                        schizophrenia || id:ieu-b-42
5                                                        schizophrenia || id:ieu-b-42
6                                      Coronary artery disease || id:ebi-a-GCST003116
7                                                      Prostate cancer || id:ieu-b-85
8                                      Coronary artery disease || id:ebi-a-GCST003116
9                                                   Alzheimer's disease || id:ieu-b-2
10                                                    Ovarian cancer || id:ieu-a-1120
11                                  High grade serous ovarian cancer || id:ieu-a-1121
12     Breast cancer (Combined Oncoarray; iCOGS; GWAS meta analysis) || id:ieu-a-1126
13                    High grade and low grade serous ovarian cancer || id:ieu-a-1228
14                                                       schizophrenia || id:ieu-b-42
15                                                  multiple sclerosis || id:ieu-b-18
16                                                  multiple sclerosis || id:ieu-b-18
17                                                     Type 2 diabetes || id:ieu-a-26
18                                                  multiple sclerosis || id:ieu-b-18
19                                                       schizophrenia || id:ieu-b-42
20                                                  Parkinson's disease || id:ieu-b-7
21                                                       schizophrenia || id:ieu-b-42
22                                                  multiple sclerosis || id:ieu-b-18
23     Breast cancer (Combined Oncoarray; iCOGS; GWAS meta analysis) || id:ieu-a-1126
24                                                    Ovarian cancer || id:ieu-a-1120
25                                  High grade serous ovarian cancer || id:ieu-a-1121
26                    High grade and low grade serous ovarian cancer || id:ieu-a-1228
27                                                Lung adenocarcinoma || id:ieu-a-984
28 ER+ Breast cancer (Combined Oncoarray; iCOGS; GWAS meta analysis) || id:ieu-a-1127
29                                                  multiple sclerosis || id:ieu-b-18
30                                                  Parkinson's disease || id:ieu-b-7
31     Breast cancer (Combined Oncoarray; iCOGS; GWAS meta analysis) || id:ieu-a-1126
32 ER+ Breast cancer (Combined Oncoarray; iCOGS; GWAS meta analysis) || id:ieu-a-1127
33                                     Coronary artery disease || id:ebi-a-GCST003116
34                                   Oral cavity and pharyngeal cancer || id:ieu-b-89
35                                     Coronary artery disease || id:ebi-a-GCST003116
36                                                  Alzheimer's disease || id:ieu-b-2
37                                                  multiple sclerosis || id:ieu-b-18
38                                                       schizophrenia || id:ieu-b-42
39     Breast cancer (Combined Oncoarray; iCOGS; GWAS meta analysis) || id:ieu-a-1126
40                                          Endometrial cancer || id:ebi-a-GCST006464
41                                                    Ovarian cancer || id:ieu-a-1120
42                    High grade and low grade serous ovarian cancer || id:ieu-a-1228
43     Breast cancer (Combined Oncoarray; iCOGS; GWAS meta analysis) || id:ieu-a-1126
44 ER+ Breast cancer (Combined Oncoarray; iCOGS; GWAS meta analysis) || id:ieu-a-1127
45                                                     Prostate cancer || id:ieu-b-85
46                                                  Alzheimer's disease || id:ieu-b-2
47                                   Oral cavity and pharyngeal cancer || id:ieu-b-89
48                                                  multiple sclerosis || id:ieu-b-18
49     Breast cancer (Combined Oncoarray; iCOGS; GWAS meta analysis) || id:ieu-a-1126
50                                                    Ovarian cancer || id:ieu-a-1120
51                                  High grade serous ovarian cancer || id:ieu-a-1121
52                    High grade and low grade serous ovarian cancer || id:ieu-a-1228
53                                                  Alzheimer's disease || id:ieu-b-2
54                                     Coronary artery disease || id:ebi-a-GCST003116
55                                                  multiple sclerosis || id:ieu-b-18
56                                                  multiple sclerosis || id:ieu-b-18
57                                                    Ovarian cancer || id:ieu-a-1120
58                                  High grade serous ovarian cancer || id:ieu-a-1121
59     Breast cancer (Combined Oncoarray; iCOGS; GWAS meta analysis) || id:ieu-a-1126
60                    High grade and low grade serous ovarian cancer || id:ieu-a-1228
61                                                  multiple sclerosis || id:ieu-b-18
62                                  High grade serous ovarian cancer || id:ieu-a-1121
63     Breast cancer (Combined Oncoarray; iCOGS; GWAS meta analysis) || id:ieu-a-1126
64                                                    Ovarian cancer || id:ieu-a-1120
65                                  High grade serous ovarian cancer || id:ieu-a-1121
66     Breast cancer (Combined Oncoarray; iCOGS; GWAS meta analysis) || id:ieu-a-1126
67 ER+ Breast cancer (Combined Oncoarray; iCOGS; GWAS meta analysis) || id:ieu-a-1127
68                    High grade and low grade serous ovarian cancer || id:ieu-a-1228
69                                                     Type 2 diabetes || id:ieu-a-26
70                                     Coronary artery disease || id:ebi-a-GCST003116
71                                            Chronic kidney disease || id:ieu-a-1102
72                                                     Prostate cancer || id:ieu-b-85
73                                                      Stroke || id:ebi-a-GCST005838
74                                                  Alzheimer's disease || id:ieu-b-2
75                                   Oral cavity and pharyngeal cancer || id:ieu-b-89
76                                            Chronic kidney disease || id:ieu-a-1102
77     Breast cancer (Combined Oncoarray; iCOGS; GWAS meta analysis) || id:ieu-a-1126
78 ER+ Breast cancer (Combined Oncoarray; iCOGS; GWAS meta analysis) || id:ieu-a-1127
   exposure                    method            Q Q_df       Q_pval
1    ADAM22 Inverse variance weighted  4.348022287    5 5.004712e-01
2       AMN Inverse variance weighted  0.991210097    3 8.033788e-01
3      ART3 Inverse variance weighted  1.028120123    3 7.944482e-01
4    BTN2A1 Inverse variance weighted 11.566856696    5 4.122982e-02
5    BTN3A2 Inverse variance weighted  3.044987252    2 2.181672e-01
6      CA11 Inverse variance weighted  4.360587808    1 3.677961e-02
7     CD160 Inverse variance weighted 14.916718429   13 3.125807e-01
8     CD207 Inverse variance weighted 15.640101871   14 3.358432e-01
9      CD33 Inverse variance weighted  1.222067836    4 8.744521e-01
10     CD34 Inverse variance weighted  3.404252826    2 1.822955e-01
11     CD34 Inverse variance weighted  2.601426871    2 2.723374e-01
12     CD34 Inverse variance weighted  2.058783562    1 1.513309e-01
13     CD34 Inverse variance weighted  3.398189439    2 1.828490e-01
14     CD40 Inverse variance weighted  8.414402364    9 4.929829e-01
15     CDH1 Inverse variance weighted  1.836330847    4 7.658291e-01
16    CDH15 Inverse variance weighted  1.094236146    2 5.786149e-01
17     CDH2 Inverse variance weighted  4.732848661    9 8.569471e-01
18     CDSN Inverse variance weighted 21.547465505   12 4.291592e-02
19   CGREF1 Inverse variance weighted  3.344621438    3 3.414803e-01
20    CRIP2 Inverse variance weighted  0.011510227    1 9.145623e-01
21    CXADR Inverse variance weighted  0.641796692    2 7.254970e-01
22    DKKL1 Inverse variance weighted  7.858621658    3 4.902574e-02
23    EDIL3 Inverse variance weighted  0.114630841    1 7.349324e-01
24    EPHB4 Inverse variance weighted  7.466569845    9 5.886618e-01
25    EPHB4 Inverse variance weighted 13.649573100    9 1.353480e-01
26    EPHB4 Inverse variance weighted  9.734509256    9 3.724027e-01
27    ERBB4 Inverse variance weighted 14.600463992   12 2.640161e-01
28    ERBIN Inverse variance weighted  3.960447002    2 1.380384e-01
29       F3 Inverse variance weighted  2.988981009    6 8.102286e-01
30   FCGR2B Inverse variance weighted  1.651582081    2 4.378885e-01
31    FGF21 Inverse variance weighted  6.122401549    6 4.096193e-01
32    FURIN Inverse variance weighted  1.828083182    2 4.009007e-01
33      GRN Inverse variance weighted 46.117036153    9 5.728204e-07
34   IFNGR1 Inverse variance weighted  8.173964734    6 2.256333e-01
35      IL6 Inverse variance weighted  2.652464185    2 2.654757e-01
36   JCHAIN Inverse variance weighted 11.876311706    9 2.203724e-01
37  KIR2DL3 Inverse variance weighted 11.285113707    2 3.543796e-03
38  KIR2DL3 Inverse variance weighted  1.399964633    3 7.055430e-01
39     LAYN Inverse variance weighted  4.689928463    3 1.959620e-01
40   LGALS4 Inverse variance weighted  0.845612635    2 6.552055e-01
41     LIFR Inverse variance weighted 19.613915401   12 7.475101e-02
42     LIFR Inverse variance weighted 21.623445376   12 4.196328e-02
43    MED18 Inverse variance weighted  0.002471217    1 9.603524e-01
44    MED18 Inverse variance weighted  0.264038760    1 6.073590e-01
45     MSMB Inverse variance weighted 18.104067608    4 1.177621e-03
46     MZB1 Inverse variance weighted  8.114389112    7 3.226154e-01
47   NOTCH1 Inverse variance weighted  4.373614265    4 3.577977e-01
48    NTRK2 Inverse variance weighted  3.622212701    3 3.052540e-01
49   PCDH17 Inverse variance weighted  4.632928711    6 5.916775e-01
50   PECAM1 Inverse variance weighted  8.678444598    6 1.924860e-01
51   PECAM1 Inverse variance weighted  9.340270053    6 1.553264e-01
52   PECAM1 Inverse variance weighted 11.460477919    6 7.514557e-02
53    PILRA Inverse variance weighted  5.434199353    5 3.652151e-01
54   PLA2G7 Inverse variance weighted 44.166810650    7 1.983898e-07
55     PLAT Inverse variance weighted  1.389987464    3 7.078834e-01
56   PLXNB2 Inverse variance weighted  9.591445825    7 2.129318e-01
57    PTPRM Inverse variance weighted 14.097806464    7 4.946887e-02
58    PTPRM Inverse variance weighted  9.050143727    7 2.490760e-01
59    PTPRM Inverse variance weighted  8.252773146    6 2.201633e-01
60    PTPRM Inverse variance weighted 12.020697445    7 9.988027e-02
61     SCGN Inverse variance weighted  6.509267683    4 1.642074e-01
62     SELE Inverse variance weighted 16.731816414    7 1.920977e-02
63     SELE Inverse variance weighted  6.976449321    6 3.230307e-01
64   SEMA4C Inverse variance weighted  8.959204156    4 6.212731e-02
65   SEMA4C Inverse variance weighted  2.975591131    4 5.619184e-01
66   SEMA4C Inverse variance weighted  5.247927381    3 1.545172e-01
67   SEMA4C Inverse variance weighted  4.362934279    3 2.248469e-01
68   SEMA4C Inverse variance weighted  4.095143346    4 3.932827e-01
69  SLITRK6 Inverse variance weighted  0.354340781    2 8.376370e-01
70    TGFB1 Inverse variance weighted  9.065501190    9 4.312500e-01
71 TNFRSF21 Inverse variance weighted  3.557295022    4 4.692202e-01
72 TNFRSF6B Inverse variance weighted  7.620032980    3 5.455319e-02
73    TRAF2 Inverse variance weighted  1.713353531    2 4.245707e-01
74    TREM2 Inverse variance weighted 13.791629830    6 3.205239e-02
75   TRIM21 Inverse variance weighted  0.280941725    1 5.960846e-01
76     UMOD Inverse variance weighted 73.631903004   11 2.478625e-11
77    VSIG4 Inverse variance weighted  6.886847700    6 3.314384e-01
78    VSIG4 Inverse variance weighted  5.962882804    6 4.273608e-01</code></pre>
</div>
</div>
<p>For multi-instrument exposures do the MR using other methods, then test if the different methods are heterogeneous</p>
<div class="cell" data-hash="all_cis_mr_cache/html/unnamed-chunk-24_4d62832ec0ab4a0b9e8f0f20018070d7">
<div class="sourceCode cell-code" id="cb119"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb119-1"><a href="#cb119-1" aria-hidden="true" tabindex="-1"></a>res2 <span class="ot">&lt;-</span> dat <span class="sc">%&gt;%</span></span>
<span id="cb119-2"><a href="#cb119-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">subset</span>(., <span class="fu">paste</span>(id.exposure, id.outcome) <span class="sc">%in%</span> <span class="fu">paste</span>(ressig<span class="sc">$</span>id.exposure, ressig<span class="sc">$</span>id.outcome)) <span class="sc">%&gt;%</span></span>
<span id="cb119-3"><a href="#cb119-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mr</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'ADAM22' on 'ieu-b-42'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'AIF1' on 'ieu-b-18'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'AIF1' on 'ieu-b-42'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'AMN' on 'ieu-b-85'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'ART3' on 'ebi-a-GCST005647'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'ATP6AP2' on 'ieu-b-42'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'ATP6V1D' on 'ieu-b-18'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'BTN2A1' on 'ieu-b-42'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'BTN3A2' on 'ieu-b-42'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'CA11' on 'ebi-a-GCST003116'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'CD160' on 'ieu-b-85'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'CD207' on 'ebi-a-GCST003116'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'CD33' on 'ieu-b-2'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'CD34' on 'ieu-a-1120'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'CD34' on 'ieu-a-1121'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'CD34' on 'ieu-a-1126'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'CD34' on 'ieu-a-1228'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'CD40' on 'ieu-b-42'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'CDH1' on 'ieu-b-18'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'CDH15' on 'ieu-b-18'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'CDH2' on 'ieu-a-26'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'CDSN' on 'ieu-b-18'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'CGREF1' on 'ieu-b-42'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'CRIP2' on 'ieu-b-7'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'CXADR' on 'ieu-b-42'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'DCN' on 'ieu-b-7'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'DCTN6' on 'ieu-b-85'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'DKKL1' on 'ieu-b-18'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'DNPH1' on 'ieu-a-1126'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'DNPH1' on 'ieu-a-1127'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'EDIL3' on 'ieu-a-1126'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'EPHB4' on 'ieu-a-1120'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'EPHB4' on 'ieu-a-1121'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'EPHB4' on 'ieu-a-1228'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'EPHB6' on 'ieu-a-1102'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'ERBB4' on 'ieu-a-984'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'ERBIN' on 'ieu-a-1127'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'F3' on 'ieu-b-18'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'FABP4' on 'ieu-a-26'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'FAM3C' on 'ieu-a-1102'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'FCGR2B' on 'ieu-b-7'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'FGF21' on 'ieu-a-1126'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'FOXO3' on 'ieu-a-989'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'FOXO3' on 'ieu-b-18'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'FOXO3' on 'ieu-b-89'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'FURIN' on 'ieu-a-1127'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'GRN' on 'ebi-a-GCST003116'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'IFNGR1' on 'ieu-b-89'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'IL6' on 'ebi-a-GCST003116'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'JCHAIN' on 'ieu-b-2'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'KIR2DL3' on 'ieu-b-18'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'KIR2DL3' on 'ieu-b-42'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'KLK10' on 'ieu-a-989'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'KLK11' on 'ieu-a-1102'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'KLK6' on 'ieu-a-1102'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'LAYN' on 'ieu-a-1126'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'LEP' on 'ieu-a-1126'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'LEP' on 'ieu-a-1127'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'LEP' on 'ieu-a-26'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'LGALS4' on 'ebi-a-GCST006464'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'LIFR' on 'ieu-a-1120'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'LIFR' on 'ieu-a-1228'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'MED18' on 'ieu-a-1126'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'MED18' on 'ieu-a-1127'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'MSMB' on 'ieu-b-85'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'MZB1' on 'ieu-b-2'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'NOTCH1' on 'ieu-b-89'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'NTRK2' on 'ieu-b-18'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'PCDH17' on 'ieu-a-1126'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'PECAM1' on 'ieu-a-1120'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'PECAM1' on 'ieu-a-1121'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'PECAM1' on 'ieu-a-1228'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'PILRA' on 'ieu-b-2'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'PLA2G7' on 'ebi-a-GCST003116'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'PLAT' on 'ieu-b-18'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'PLXNB2' on 'ieu-b-18'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'POLR2F' on 'ebi-a-GCST003116'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'POLR2F' on 'ebi-a-GCST005838'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'POLR2F' on 'ebi-a-GCST006464'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'POLR2F' on 'ieu-b-18'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'PTPRM' on 'ieu-a-1120'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'PTPRM' on 'ieu-a-1121'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'PTPRM' on 'ieu-a-1126'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'PTPRM' on 'ieu-a-1228'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'PVR' on 'ieu-b-18'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'PVR' on 'ieu-b-2'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'SCGN' on 'ieu-b-18'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'SELE' on 'ieu-a-1121'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'SELE' on 'ieu-a-1126'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'SEMA4C' on 'ieu-a-1120'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'SEMA4C' on 'ieu-a-1121'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'SEMA4C' on 'ieu-a-1126'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'SEMA4C' on 'ieu-a-1127'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'SEMA4C' on 'ieu-a-1228'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'SIGLEC5' on 'ieu-b-18'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'SLITRK6' on 'ieu-a-26'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'TFF3' on 'ieu-a-1102'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'TGFB1' on 'ebi-a-GCST003116'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'TNFRSF10A' on 'ieu-b-18'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'TNFRSF21' on 'ieu-a-1102'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'TNFRSF6B' on 'ieu-b-85'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'TRAF2' on 'ebi-a-GCST005838'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'TREM2' on 'ieu-b-2'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'TRIM21' on 'ieu-b-89'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'UMOD' on 'ieu-a-1102'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'VSIG4' on 'ieu-a-1126'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'VSIG4' on 'ieu-a-1127'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Analysing 'XG' on 'ieu-a-26'</code></pre>
</div>
<div class="sourceCode cell-code" id="cb228"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb228-1"><a href="#cb228-1" aria-hidden="true" tabindex="-1"></a>fn <span class="ot">&lt;-</span> <span class="cf">function</span>(b, se)</span>
<span id="cb228-2"><a href="#cb228-2" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb228-3"><a href="#cb228-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rma.uni</span>(b, <span class="at">sei=</span>se, <span class="at">method=</span><span class="st">"FE"</span>)<span class="sc">$</span>QEp</span>
<span id="cb228-4"><a href="#cb228-4" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb228-5"><a href="#cb228-5" aria-hidden="true" tabindex="-1"></a>het_method <span class="ot">&lt;-</span> res2 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(nsnp <span class="sc">&gt;=</span> <span class="dv">3</span>) <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(id.exposure, id.outcome) <span class="sc">%&gt;%</span></span>
<span id="cb228-6"><a href="#cb228-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">do</span>({</span>
<span id="cb228-7"><a href="#cb228-7" aria-hidden="true" tabindex="-1"></a>        x <span class="ot">&lt;-</span> .</span>
<span id="cb228-8"><a href="#cb228-8" aria-hidden="true" tabindex="-1"></a>        <span class="fu">tibble</span>(<span class="at">method_Q_pval=</span><span class="fu">fn</span>(x<span class="sc">$</span>b, x<span class="sc">$</span>se))</span>
<span id="cb228-9"><a href="#cb228-9" aria-hidden="true" tabindex="-1"></a>    })</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Plot comparison across different methods</p>
<div class="cell" data-hash="all_cis_mr_cache/html/unnamed-chunk-25_f572e898c96932da345aef4c369e6833">
<div class="sourceCode cell-code" id="cb229"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb229-1"><a href="#cb229-1" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> res <span class="sc">%&gt;%</span> </span>
<span id="cb229-2"><a href="#cb229-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(nsnp <span class="sc">&gt;</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb229-3"><a href="#cb229-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="fu">p.adjust</span>(pval, <span class="st">"fdr"</span>) <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span></span>
<span id="cb229-4"><a href="#cb229-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">code =</span> <span class="fu">paste</span>(id.exposure, id.outcome)) <span class="sc">%&gt;%</span> {.<span class="sc">$</span>code}</span>
<span id="cb229-5"><a href="#cb229-5" aria-hidden="true" tabindex="-1"></a>res2 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">paste</span>(id.exposure, id.outcome) <span class="sc">%in%</span> p) <span class="sc">%&gt;%</span></span>
<span id="cb229-6"><a href="#cb229-6" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(., <span class="fu">aes</span>(<span class="at">x=</span>method, <span class="at">y=</span>b)) <span class="sc">+</span></span>
<span id="cb229-7"><a href="#cb229-7" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">colour=</span>method)) <span class="sc">+</span></span>
<span id="cb229-8"><a href="#cb229-8" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_errorbar</span>(<span class="fu">aes</span>(<span class="at">ymin=</span>b<span class="sc">-</span>se<span class="sc">*</span><span class="fl">1.96</span>, <span class="at">ymax=</span>b<span class="sc">+</span>se<span class="sc">*</span><span class="fl">1.96</span>, <span class="at">colour=</span>method), <span class="at">width=</span><span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb229-9"><a href="#cb229-9" aria-hidden="true" tabindex="-1"></a><span class="fu">facet_wrap</span>(<span class="sc">~</span> id.exposure<span class="sc">+</span>id.outcome, <span class="at">scale=</span><span class="st">"free_y"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="all_cis_mr_files/figure-html/unnamed-chunk-25-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>For multi-instrument exposures do leave-one-out analysis to see if any are driven by a single SNP</p>
<div class="cell" data-hash="all_cis_mr_cache/html/unnamed-chunk-26_400d3093ec7ed3921683c0a817e042a9">
<div class="sourceCode cell-code" id="cb230"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb230-1"><a href="#cb230-1" aria-hidden="true" tabindex="-1"></a>mr_leaveoneout2 <span class="ot">&lt;-</span> <span class="cf">function</span> (dat, <span class="at">parameters =</span> <span class="fu">default_parameters</span>(), <span class="at">method =</span> mr_ivw) {</span>
<span id="cb230-2"><a href="#cb230-2" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="st">"samplesize.outcome"</span> <span class="sc">%in%</span> <span class="fu">names</span>(dat)) {</span>
<span id="cb230-3"><a href="#cb230-3" aria-hidden="true" tabindex="-1"></a>        dat<span class="sc">$</span>samplesize.outcome <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb230-4"><a href="#cb230-4" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb230-5"><a href="#cb230-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stopifnot</span>(<span class="st">"outcome"</span> <span class="sc">%in%</span> <span class="fu">names</span>(dat))</span>
<span id="cb230-6"><a href="#cb230-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stopifnot</span>(<span class="st">"exposure"</span> <span class="sc">%in%</span> <span class="fu">names</span>(dat))</span>
<span id="cb230-7"><a href="#cb230-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stopifnot</span>(<span class="st">"beta.exposure"</span> <span class="sc">%in%</span> <span class="fu">names</span>(dat))</span>
<span id="cb230-8"><a href="#cb230-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stopifnot</span>(<span class="st">"beta.outcome"</span> <span class="sc">%in%</span> <span class="fu">names</span>(dat))</span>
<span id="cb230-9"><a href="#cb230-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stopifnot</span>(<span class="st">"se.exposure"</span> <span class="sc">%in%</span> <span class="fu">names</span>(dat))</span>
<span id="cb230-10"><a href="#cb230-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stopifnot</span>(<span class="st">"se.outcome"</span> <span class="sc">%in%</span> <span class="fu">names</span>(dat))</span>
<span id="cb230-11"><a href="#cb230-11" aria-hidden="true" tabindex="-1"></a>    res <span class="ot">&lt;-</span> plyr<span class="sc">::</span><span class="fu">ddply</span>(dat, <span class="fu">c</span>(<span class="st">"id.exposure"</span>, <span class="st">"id.outcome"</span>), <span class="cf">function</span>(X) {</span>
<span id="cb230-12"><a href="#cb230-12" aria-hidden="true" tabindex="-1"></a>        x <span class="ot">&lt;-</span> <span class="fu">subset</span>(X, mr_keep)</span>
<span id="cb230-13"><a href="#cb230-13" aria-hidden="true" tabindex="-1"></a>        nsnp <span class="ot">&lt;-</span> <span class="fu">nrow</span>(x)</span>
<span id="cb230-14"><a href="#cb230-14" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (nsnp <span class="sc">==</span> <span class="dv">0</span>) {</span>
<span id="cb230-15"><a href="#cb230-15" aria-hidden="true" tabindex="-1"></a>            x <span class="ot">&lt;-</span> X[<span class="dv">1</span>, ]</span>
<span id="cb230-16"><a href="#cb230-16" aria-hidden="true" tabindex="-1"></a>            d <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">SNP =</span> <span class="st">"All"</span>, <span class="at">b =</span> <span class="cn">NA</span>, <span class="at">se =</span> <span class="cn">NA</span>, <span class="at">p =</span> <span class="cn">NA</span>, </span>
<span id="cb230-17"><a href="#cb230-17" aria-hidden="true" tabindex="-1"></a>                <span class="at">samplesize =</span> <span class="cn">NA</span>, <span class="at">outcome =</span> x<span class="sc">$</span>outcome[<span class="dv">1</span>], <span class="at">exposure =</span> x<span class="sc">$</span>exposure[<span class="dv">1</span>])</span>
<span id="cb230-18"><a href="#cb230-18" aria-hidden="true" tabindex="-1"></a>            <span class="fu">return</span>(d)</span>
<span id="cb230-19"><a href="#cb230-19" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb230-20"><a href="#cb230-20" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (nsnp <span class="sc">&gt;</span> <span class="dv">2</span>) {</span>
<span id="cb230-21"><a href="#cb230-21" aria-hidden="true" tabindex="-1"></a>            l <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="dv">1</span><span class="sc">:</span>nsnp, <span class="cf">function</span>(i) {</span>
<span id="cb230-22"><a href="#cb230-22" aria-hidden="true" tabindex="-1"></a>                <span class="fu">with</span>(x, <span class="fu">method</span>(beta.exposure[<span class="sc">-</span>i], beta.outcome[<span class="sc">-</span>i], </span>
<span id="cb230-23"><a href="#cb230-23" aria-hidden="true" tabindex="-1"></a>                  se.exposure[<span class="sc">-</span>i], se.outcome[<span class="sc">-</span>i], parameters))</span>
<span id="cb230-24"><a href="#cb230-24" aria-hidden="true" tabindex="-1"></a>            })</span>
<span id="cb230-25"><a href="#cb230-25" aria-hidden="true" tabindex="-1"></a>            l[[nsnp <span class="sc">+</span> <span class="dv">1</span>]] <span class="ot">&lt;-</span> <span class="fu">with</span>(x, <span class="fu">method</span>(beta.exposure, beta.outcome, </span>
<span id="cb230-26"><a href="#cb230-26" aria-hidden="true" tabindex="-1"></a>                se.exposure, se.outcome, parameters))</span>
<span id="cb230-27"><a href="#cb230-27" aria-hidden="true" tabindex="-1"></a>            d <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">SNP =</span> <span class="fu">c</span>(<span class="fu">as.character</span>(x<span class="sc">$</span>SNP), <span class="st">"All"</span>), </span>
<span id="cb230-28"><a href="#cb230-28" aria-hidden="true" tabindex="-1"></a>                <span class="at">b =</span> <span class="fu">sapply</span>(l, <span class="cf">function</span>(y) y<span class="sc">$</span>b), <span class="at">se =</span> <span class="fu">sapply</span>(l, </span>
<span id="cb230-29"><a href="#cb230-29" aria-hidden="true" tabindex="-1"></a>                  <span class="cf">function</span>(y) y<span class="sc">$</span>se), <span class="at">p =</span> <span class="fu">sapply</span>(l, <span class="cf">function</span>(y) y<span class="sc">$</span>pval), </span>
<span id="cb230-30"><a href="#cb230-30" aria-hidden="true" tabindex="-1"></a>                <span class="at">samplesize =</span> x<span class="sc">$</span>samplesize.outcome[<span class="dv">1</span>])</span>
<span id="cb230-31"><a href="#cb230-31" aria-hidden="true" tabindex="-1"></a>            d<span class="sc">$</span>outcome <span class="ot">&lt;-</span> x<span class="sc">$</span>outcome[<span class="dv">1</span>]</span>
<span id="cb230-32"><a href="#cb230-32" aria-hidden="true" tabindex="-1"></a>            d<span class="sc">$</span>exposure <span class="ot">&lt;-</span> x<span class="sc">$</span>exposure[<span class="dv">1</span>]</span>
<span id="cb230-33"><a href="#cb230-33" aria-hidden="true" tabindex="-1"></a>        } <span class="cf">else</span> <span class="cf">if</span>(nsnp <span class="sc">==</span> <span class="dv">2</span>) {</span>
<span id="cb230-34"><a href="#cb230-34" aria-hidden="true" tabindex="-1"></a>            l <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="dv">1</span><span class="sc">:</span>nsnp, <span class="cf">function</span>(i) {</span>
<span id="cb230-35"><a href="#cb230-35" aria-hidden="true" tabindex="-1"></a>                <span class="fu">with</span>(x, <span class="fu">mr_wald_ratio</span>(beta.exposure[<span class="sc">-</span>i], beta.outcome[<span class="sc">-</span>i], </span>
<span id="cb230-36"><a href="#cb230-36" aria-hidden="true" tabindex="-1"></a>                  se.exposure[<span class="sc">-</span>i], se.outcome[<span class="sc">-</span>i], parameters))</span>
<span id="cb230-37"><a href="#cb230-37" aria-hidden="true" tabindex="-1"></a>            })</span>
<span id="cb230-38"><a href="#cb230-38" aria-hidden="true" tabindex="-1"></a>            l[[nsnp <span class="sc">+</span> <span class="dv">1</span>]] <span class="ot">&lt;-</span> <span class="fu">with</span>(x, <span class="fu">method</span>(beta.exposure, beta.outcome, </span>
<span id="cb230-39"><a href="#cb230-39" aria-hidden="true" tabindex="-1"></a>                se.exposure, se.outcome, parameters))</span>
<span id="cb230-40"><a href="#cb230-40" aria-hidden="true" tabindex="-1"></a>            d <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">SNP =</span> <span class="fu">c</span>(<span class="fu">as.character</span>(x<span class="sc">$</span>SNP), <span class="st">"All"</span>), </span>
<span id="cb230-41"><a href="#cb230-41" aria-hidden="true" tabindex="-1"></a>                <span class="at">b =</span> <span class="fu">sapply</span>(l, <span class="cf">function</span>(y) y<span class="sc">$</span>b), <span class="at">se =</span> <span class="fu">sapply</span>(l, </span>
<span id="cb230-42"><a href="#cb230-42" aria-hidden="true" tabindex="-1"></a>                  <span class="cf">function</span>(y) y<span class="sc">$</span>se), <span class="at">p =</span> <span class="fu">sapply</span>(l, <span class="cf">function</span>(y) y<span class="sc">$</span>pval), </span>
<span id="cb230-43"><a href="#cb230-43" aria-hidden="true" tabindex="-1"></a>                <span class="at">samplesize =</span> x<span class="sc">$</span>samplesize.outcome[<span class="dv">1</span>])</span>
<span id="cb230-44"><a href="#cb230-44" aria-hidden="true" tabindex="-1"></a>            d<span class="sc">$</span>outcome <span class="ot">&lt;-</span> x<span class="sc">$</span>outcome[<span class="dv">1</span>]</span>
<span id="cb230-45"><a href="#cb230-45" aria-hidden="true" tabindex="-1"></a>            d<span class="sc">$</span>exposure <span class="ot">&lt;-</span> x<span class="sc">$</span>exposure[<span class="dv">1</span>]</span>
<span id="cb230-46"><a href="#cb230-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb230-47"><a href="#cb230-47" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb230-48"><a href="#cb230-48" aria-hidden="true" tabindex="-1"></a>        <span class="cf">else</span> {</span>
<span id="cb230-49"><a href="#cb230-49" aria-hidden="true" tabindex="-1"></a>            a <span class="ot">&lt;-</span> <span class="fu">with</span>(x, <span class="fu">method</span>(beta.exposure, beta.outcome, </span>
<span id="cb230-50"><a href="#cb230-50" aria-hidden="true" tabindex="-1"></a>                se.exposure, se.outcome, parameters))</span>
<span id="cb230-51"><a href="#cb230-51" aria-hidden="true" tabindex="-1"></a>            d <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">SNP =</span> <span class="st">"All"</span>, <span class="at">b =</span> a<span class="sc">$</span>b, <span class="at">se =</span> a<span class="sc">$</span>se, </span>
<span id="cb230-52"><a href="#cb230-52" aria-hidden="true" tabindex="-1"></a>                <span class="at">p =</span> a<span class="sc">$</span>pval, <span class="at">samplesize =</span> x<span class="sc">$</span>samplesize.outcome[<span class="dv">1</span>])</span>
<span id="cb230-53"><a href="#cb230-53" aria-hidden="true" tabindex="-1"></a>            d<span class="sc">$</span>outcome <span class="ot">&lt;-</span> x<span class="sc">$</span>outcome[<span class="dv">1</span>]</span>
<span id="cb230-54"><a href="#cb230-54" aria-hidden="true" tabindex="-1"></a>            d<span class="sc">$</span>exposure <span class="ot">&lt;-</span> x<span class="sc">$</span>exposure[<span class="dv">1</span>]</span>
<span id="cb230-55"><a href="#cb230-55" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb230-56"><a href="#cb230-56" aria-hidden="true" tabindex="-1"></a>        <span class="fu">return</span>(d)</span>
<span id="cb230-57"><a href="#cb230-57" aria-hidden="true" tabindex="-1"></a>    })</span>
<span id="cb230-58"><a href="#cb230-58" aria-hidden="true" tabindex="-1"></a>    res <span class="ot">&lt;-</span> <span class="fu">subset</span>(res, <span class="at">select =</span> <span class="fu">c</span>(exposure, outcome, id.exposure, </span>
<span id="cb230-59"><a href="#cb230-59" aria-hidden="true" tabindex="-1"></a>        id.outcome, samplesize, SNP, b, se, p))</span>
<span id="cb230-60"><a href="#cb230-60" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(res)</span>
<span id="cb230-61"><a href="#cb230-61" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb230-62"><a href="#cb230-62" aria-hidden="true" tabindex="-1"></a>loo <span class="ot">&lt;-</span> dat <span class="sc">%&gt;%</span></span>
<span id="cb230-63"><a href="#cb230-63" aria-hidden="true" tabindex="-1"></a>    <span class="fu">subset</span>(., <span class="fu">paste</span>(id.exposure, id.outcome) <span class="sc">%in%</span> <span class="fu">paste</span>(ressig<span class="sc">$</span>id.exposure, ressig<span class="sc">$</span>id.outcome)) <span class="sc">%&gt;%</span></span>
<span id="cb230-64"><a href="#cb230-64" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mr_leaveoneout2</span>(.)</span>
<span id="cb230-65"><a href="#cb230-65" aria-hidden="true" tabindex="-1"></a>loo_passing <span class="ot">&lt;-</span> loo <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(id.exposure, id.outcome) <span class="sc">%&gt;%</span></span>
<span id="cb230-66"><a href="#cb230-66" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(SNP <span class="sc">!=</span> <span class="st">"All"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb230-67"><a href="#cb230-67" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">maxp=</span><span class="fu">max</span>(p, <span class="at">na.rm=</span>T)) <span class="sc">%&gt;%</span></span>
<span id="cb230-68"><a href="#cb230-68" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(maxp)) <span class="sc">%&gt;%</span> </span>
<span id="cb230-69"><a href="#cb230-69" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">fdr=</span><span class="fu">p.adjust</span>(maxp, <span class="st">"fdr"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'id.exposure'. You can override using the
`.groups` argument.</code></pre>
</div>
</div>
<p>Arrange all results</p>
<div class="cell" data-hash="all_cis_mr_cache/html/unnamed-chunk-27_ebf0dab6500f86ef75eea032287f0bbb">
<div class="sourceCode cell-code" id="cb232"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb232-1"><a href="#cb232-1" aria-hidden="true" tabindex="-1"></a>allres <span class="ot">&lt;-</span> <span class="fu">left_join</span>(</span>
<span id="cb232-2"><a href="#cb232-2" aria-hidden="true" tabindex="-1"></a>    ressig <span class="sc">%&gt;%</span> <span class="fu">select</span>(id.exposure, id.outcome, nsnp, b, se, pval), </span>
<span id="cb232-3"><a href="#cb232-3" aria-hidden="true" tabindex="-1"></a>    loo_passing <span class="sc">%&gt;%</span> <span class="fu">select</span>(id.exposure, id.outcome, <span class="at">loo_maxp=</span>maxp, <span class="at">loo_fdr=</span>fdr)) <span class="sc">%&gt;%</span> </span>
<span id="cb232-4"><a href="#cb232-4" aria-hidden="true" tabindex="-1"></a><span class="fu">left_join</span>(., het2 <span class="sc">%&gt;%</span> <span class="fu">select</span>(id.exposure, id.outcome, <span class="at">inst_Q_pval=</span>Q_pval)) <span class="sc">%&gt;%</span></span>
<span id="cb232-5"><a href="#cb232-5" aria-hidden="true" tabindex="-1"></a><span class="fu">left_join</span>(., het_method) <span class="sc">%&gt;%</span></span>
<span id="cb232-6"><a href="#cb232-6" aria-hidden="true" tabindex="-1"></a><span class="fu">arrange</span>(loo_maxp,  <span class="fu">desc</span>(inst_Q_pval)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">duplicated</span>(<span class="fu">paste</span>(id.exposure, id.outcome)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining with `by = join_by(id.exposure, id.outcome)`
Joining with `by = join_by(id.exposure, id.outcome)`
Joining with `by = join_by(id.exposure, id.outcome)`</code></pre>
</div>
<div class="sourceCode cell-code" id="cb234"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb234-1"><a href="#cb234-1" aria-hidden="true" tabindex="-1"></a><span class="co"># subset(ressig, nsnp %in% 2:3) %&gt;% </span></span>
<span id="cb234-2"><a href="#cb234-2" aria-hidden="true" tabindex="-1"></a><span class="co">#     select(id.exposure, id.outcome, nsnp, b, se, pval) %&gt;% </span></span>
<span id="cb234-3"><a href="#cb234-3" aria-hidden="true" tabindex="-1"></a><span class="co">#     inner_join(., het2 %&gt;% filter(method=="Inverse variance weighted") %&gt;% select(id.exposure, id.outcome, inst_Q_pval=Q_pval)) %&gt;% </span></span>
<span id="cb234-4"><a href="#cb234-4" aria-hidden="true" tabindex="-1"></a><span class="co">#     left_join(., loo_passing %&gt;% select(id.exposure, id.outcome, loo_maxp=maxp)) %&gt;%</span></span>
<span id="cb234-5"><a href="#cb234-5" aria-hidden="true" tabindex="-1"></a><span class="co">#     arrange(loo_maxp, desc(inst_Q_pval))</span></span>
<span id="cb234-6"><a href="#cb234-6" aria-hidden="true" tabindex="-1"></a><span class="co"># subset(ressig, nsnp==1) %&gt;% arrange(pval)</span></span>
<span id="cb234-7"><a href="#cb234-7" aria-hidden="true" tabindex="-1"></a>traits <span class="ot">&lt;-</span> readxl<span class="sc">::</span><span class="fu">read_xlsx</span>(<span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"Supplementary_Table5.xlsx"</span>)) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="at">id.outcome=</span>opengwasid, <span class="at">outcome=</span>code)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>New names:
• `` -&gt; `...2`</code></pre>
</div>
<div class="sourceCode cell-code" id="cb236"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb236-1"><a href="#cb236-1" aria-hidden="true" tabindex="-1"></a>m <span class="ot">&lt;-</span> <span class="fu">match</span>(allres<span class="sc">$</span>id.outcome, traits<span class="sc">$</span>id.outcome)</span>
<span id="cb236-2"><a href="#cb236-2" aria-hidden="true" tabindex="-1"></a><span class="fu">all</span>(traits<span class="sc">$</span>id.outcome[m] <span class="sc">==</span> allres<span class="sc">$</span>id.outcome)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE</code></pre>
</div>
<div class="sourceCode cell-code" id="cb238"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb238-1"><a href="#cb238-1" aria-hidden="true" tabindex="-1"></a>allres<span class="sc">$</span>id.outcome <span class="ot">&lt;-</span> traits<span class="sc">$</span>outcome[m]</span>
<span id="cb238-2"><a href="#cb238-2" aria-hidden="true" tabindex="-1"></a>allres</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>    id.exposure
1        JCHAIN
2        PLA2G7
3         TGFB1
4         MED18
5           IL6
6          MZB1
7          ART3
8        CGREF1
9          CDH1
10        MED18
11         CDH2
12       PLXNB2
13       TRIM21
14        VSIG4
15        FGF21
16        CRIP2
17           F3
18         PLAT
19     TNFRSF21
20        TRAF2
21        ERBB4
22         SCGN
23        CD160
24        NTRK2
25       ADAM22
26       PCDH17
27         LAYN
28        ERBIN
29        VSIG4
30       SEMA4C
31        CXADR
32         CD40
33       SEMA4C
34      SLITRK6
35       IFNGR1
36       BTN2A1
37        CD207
38          AMN
39       LGALS4
40      KIR2DL3
41        DKKL1
42        FURIN
43        EDIL3
44        PTPRM
45       NOTCH1
46        EPHB4
47         UMOD
48         SELE
49     TNFRSF6B
50       BTN3A2
51         CDSN
52         CA11
53        PILRA
54       SEMA4C
55       SEMA4C
56      KIR2DL3
57       SEMA4C
58        EPHB4
59        CDH15
60         CD34
61       FCGR2B
62        TREM2
63        PTPRM
64        EPHB4
65          GRN
66        PTPRM
67         LIFR
68       PECAM1
69         CD34
70         SELE
71         LIFR
72         CD34
73         CD34
74         MSMB
75         CD33
76       PECAM1
77        PTPRM
78       PECAM1
79         AIF1
80         AIF1
81      ATP6AP2
82      ATP6V1D
83          DCN
84        DCTN6
85        DNPH1
86        DNPH1
87        EPHB6
88        FABP4
89        FAM3C
90        FOXO3
91        FOXO3
92        FOXO3
93        KLK10
94        KLK11
95         KLK6
96          LEP
97          LEP
98          LEP
99       POLR2F
100      POLR2F
101      POLR2F
102      POLR2F
103         PVR
104         PVR
105     SIGLEC5
106        TFF3
107   TNFRSF10A
108          XG
                                                           id.outcome nsnp
1                                                                  AD   10
2                                                                 CAD    8
3                                                                 CAD   10
4                                                       Breast cancer    2
5                                                                 CAD    3
6                                                                  AD    8
7                                                                 ALS    4
8                                                       Schizophrenia    4
9                                                                  MS    5
10  ER+ Breast cancer (Combined Oncoarray; iCOGS; GWAS meta analysis)    2
11                                                                T2D   10
12                                                                 MS    8
13                                  Oral cavity and pharyngeal cancer    2
14                                                      Breast cancer    7
15                                                      Breast cancer    7
16                                                                 PD    2
17                                                                 MS    7
18                                                                 MS    4
19                                                                CKD    5
20                                                             Stroke    3
21                                                Lung adenocarcinoma   13
22                                                                 MS    5
23                                                    Prostate cancer   14
24                                                                 MS    4
25                                                      Schizophrenia    6
26                                                      Breast cancer    7
27                                                      Breast cancer    4
28  ER+ Breast cancer (Combined Oncoarray; iCOGS; GWAS meta analysis)    3
29  ER+ Breast cancer (Combined Oncoarray; iCOGS; GWAS meta analysis)    7
30                                   High grade serous ovarian cancer    5
31                                                      Schizophrenia    3
32                                                      Schizophrenia   10
33                     High grade and low grade serous ovarian cancer    5
34                                                                T2D    3
35                                  Oral cavity and pharyngeal cancer    7
36                                                      Schizophrenia    6
37                                                                CAD   15
38                                                    Prostate cancer    4
39                                                 Endometrial cancer    3
40                                                      Schizophrenia    4
41                                                                 MS    4
42  ER+ Breast cancer (Combined Oncoarray; iCOGS; GWAS meta analysis)    3
43                                                      Breast cancer    2
44                                                     Ovarian cancer    8
45                                  Oral cavity and pharyngeal cancer    5
46                                                     Ovarian cancer   10
47                                                                CKD   12
48                                   High grade serous ovarian cancer    8
49                                                    Prostate cancer    4
50                                                      Schizophrenia    3
51                                                                 MS   13
52                                                                CAD    2
53                                                                 AD    6
54                                                      Breast cancer    4
55                                                     Ovarian cancer    5
56                                                                 MS    3
57  ER+ Breast cancer (Combined Oncoarray; iCOGS; GWAS meta analysis)    4
58                     High grade and low grade serous ovarian cancer   10
59                                                                 MS    3
60                                                      Breast cancer    2
61                                                                 PD    3
62                                                                 AD    7
63                                                      Breast cancer    7
64                                   High grade serous ovarian cancer   10
65                                                                CAD   10
66                     High grade and low grade serous ovarian cancer    8
67                                                     Ovarian cancer   13
68                                                     Ovarian cancer    7
69                                                     Ovarian cancer    3
70                                                      Breast cancer    7
71                     High grade and low grade serous ovarian cancer   13
72                     High grade and low grade serous ovarian cancer    3
73                                   High grade serous ovarian cancer    3
74                                                    Prostate cancer    5
75                                                                 AD    5
76                     High grade and low grade serous ovarian cancer    7
77                                   High grade serous ovarian cancer    8
78                                   High grade serous ovarian cancer    7
79                                                                 MS    1
80                                                      Schizophrenia    1
81                                                      Schizophrenia    1
82                                                                 MS    1
83                                                                 PD    1
84                                                    Prostate cancer    1
85                                                      Breast cancer    1
86  ER+ Breast cancer (Combined Oncoarray; iCOGS; GWAS meta analysis)    1
87                                                                CKD    1
88                                                                T2D    1
89                                                                CKD    1
90                                          Squamous cell lung cancer    1
91                                                                 MS    1
92                                  Oral cavity and pharyngeal cancer    1
93                                          Squamous cell lung cancer    1
94                                                                CKD    1
95                                                                CKD    1
96                                                      Breast cancer    1
97  ER+ Breast cancer (Combined Oncoarray; iCOGS; GWAS meta analysis)    1
98                                                                T2D    1
99                                                                CAD    1
100                                                            Stroke    1
101                                                Endometrial cancer    1
102                                                                MS    1
103                                                                MS    1
104                                                                AD    1
105                                                                MS    1
106                                                               CKD    1
107                                                                MS    1
108                                                               T2D    1
              b          se          pval     loo_maxp      loo_fdr
1   -0.44042150 0.083697323  1.424395e-07 7.459978e-06 7.459978e-06
2    0.66290565 0.131469698  4.600427e-07 1.352275e-05 1.352275e-05
3    0.20846682 0.032276834  1.055970e-10 2.290986e-05 2.290986e-05
4   -0.41686884 0.079410636  1.524843e-07 2.999020e-04 5.998041e-04
5   -0.22031049 0.045503503  1.287874e-06 1.192762e-03 1.192762e-03
6   -0.28640684 0.071754002  6.565381e-05 1.530371e-03 1.530371e-03
7    0.48145467 0.118816338  5.076180e-05 1.541791e-03 1.541791e-03
8   -0.07610900 0.018806610  5.189415e-05 2.274879e-03 2.274879e-03
9    0.14379775 0.032787043  1.155548e-05 2.507865e-03 2.507865e-03
10  -0.46741495 0.094448714  7.464338e-07 2.586900e-03 2.586900e-03
11   0.31161255 0.084747594  2.360406e-04 3.087758e-03 3.087758e-03
12   0.52439922 0.132817778  7.872103e-05 3.389538e-03 3.389538e-03
13  -2.74614447 0.668473464  3.989567e-05 3.405236e-03 3.405236e-03
14   0.19994825 0.043651181  4.636614e-06 3.831182e-03 7.662364e-03
15  -0.10765342 0.028124415  1.293164e-04 4.898918e-03 4.898918e-03
16   0.50940780 0.127278866  6.273053e-05 5.595514e-03 5.595514e-03
17   0.17193859 0.040828454  2.539681e-05 5.596752e-03 5.596752e-03
18  -0.61412506 0.145856509  2.548353e-05 5.853478e-03 5.853478e-03
19   0.41338892 0.103167361  6.150168e-05 7.010737e-03 7.010737e-03
20   0.33135684 0.082475618  5.878543e-05 7.354920e-03 7.354920e-03
21   0.26812728 0.071230357  1.670646e-04 7.755529e-03 7.755529e-03
22   0.42018822 0.106045072  7.421407e-05 8.341515e-03 8.341515e-03
23  -0.06437167 0.017470094  2.289920e-04 9.743667e-03 9.743667e-03
24   0.69151071 0.173322341  6.614635e-05 1.018231e-02 1.018231e-02
25  -0.09481963 0.020003084  2.134543e-06 1.020694e-02 1.020694e-02
26  -0.05207741 0.012541130  3.288208e-05 1.225207e-02 1.225207e-02
27   0.07513424 0.019640460  1.305161e-04 1.287310e-02 1.287310e-02
28  -0.48689164 0.116081661  2.736081e-05 1.433583e-02 1.433583e-02
29   0.18257406 0.048428546  1.632686e-04 1.600519e-02 1.600519e-02
30  -0.46259458 0.066897012  4.677773e-12 2.297363e-02 7.971142e-02
31  -0.19650709 0.048911993  5.880274e-05 2.809875e-02 2.809875e-02
32  -0.10340262 0.026259541  8.226086e-05 3.165504e-02 3.165504e-02
33  -0.43272716 0.065921830  5.229653e-11 3.188457e-02 7.971142e-02
34   0.60381592 0.163897926  2.295145e-04 3.209092e-02 3.209092e-02
35  -0.62439325 0.164043275  1.410775e-04 3.603146e-02 3.603146e-02
36  -0.40117244 0.059474410  1.527165e-11 3.671320e-02 3.671320e-02
37   0.08056135 0.020974591  1.225832e-04 3.713765e-02 3.713765e-02
38  -0.18410099 0.038231233  1.468609e-06 4.836040e-02 4.836040e-02
39  -0.12631569 0.030199211  2.880066e-05 9.024600e-02 9.024600e-02
40  -0.09150675 0.019776174  3.707842e-06 1.100998e-01 2.201996e-01
41  -0.09162408 0.023921527  1.280424e-04 1.181965e-01 1.181965e-01
42  -0.14422806 0.034902116  3.590596e-05 1.195394e-01 1.195394e-01
43  -0.21741168 0.054818288  7.307570e-05 1.326716e-01 1.326716e-01
44  -0.12083582 0.032645112  2.143303e-04 2.308238e-01 8.870049e-01
45  -0.58715919 0.144024000  4.565681e-05 2.324787e-01 2.324787e-01
46  -0.17298176 0.034984013  7.630252e-07 2.506073e-01 6.604988e-01
47   0.23253936 0.055380732  2.681889e-05 2.665353e-01 2.665353e-01
48  -0.11986579 0.030354652  7.852946e-05 2.799803e-01 5.599606e-01
49  -0.22845727 0.042473942  7.499929e-08 3.150437e-01 3.150437e-01
50   0.16898623 0.022693785  9.594635e-14 3.356108e-01 3.356108e-01
51  -0.20081094 0.033693935  2.524631e-09 3.408986e-01 3.408986e-01
52   0.49683238 0.119972285  3.454539e-05 3.484329e-01 3.484329e-01
53   0.06037403 0.016364418  2.248293e-04 3.534379e-01 3.534379e-01
54  -0.15245508 0.036410734  2.825518e-05 4.088733e-01 4.705198e-01
55  -0.32142598 0.084840395  1.515011e-04 4.279810e-01 4.705198e-01
56   0.56374331 0.074415569  3.574447e-14 4.468358e-01 4.468358e-01
57  -0.14881616 0.039218386  1.479110e-04 4.705198e-01 4.705198e-01
58  -0.19841193 0.041881600  2.164256e-06 4.763733e-01 6.604988e-01
59   0.19790699 0.050653484  9.341992e-05 5.001589e-01 5.001589e-01
60  -0.09091735 0.024280906  1.808275e-04 5.316335e-01 8.893914e-01
61  -0.15578567 0.037514773  3.286489e-05 5.669959e-01 5.669959e-01
62  -0.22386537 0.055639672  5.734332e-05 5.804116e-01 5.804116e-01
63  -0.05744611 0.012796336  7.147026e-06 6.151465e-01 8.870049e-01
64  -0.20783712 0.050946648  4.513164e-05 6.604988e-01 6.604988e-01
65   0.12511689 0.032548294  1.210260e-04 6.638767e-01 6.638767e-01
66  -0.15705036 0.034605485  5.670732e-06 6.652536e-01 8.870049e-01
67  -0.15356868 0.040502197  1.496716e-04 7.284582e-01 8.671421e-01
68  -0.14429034 0.032388095  8.387662e-06 7.427112e-01 9.665238e-01
69  -0.19307636 0.045810827  2.501717e-05 7.550660e-01 8.893914e-01
70  -0.04283690 0.008482658  4.419522e-07 8.191592e-01 8.191592e-01
71  -0.18799036 0.048925906  1.218628e-04 8.671421e-01 8.671421e-01
72  -0.24320674 0.052541535  3.677070e-06 8.871231e-01 8.893914e-01
73  -0.26509677 0.047185705  1.930035e-08 8.893914e-01 8.893914e-01
74  -0.21918785 0.018158286  1.503800e-33 9.050100e-01 9.050100e-01
75   0.04879778 0.012533807  9.889266e-05 9.544734e-01 9.544734e-01
76  -0.18330424 0.042741006  1.797051e-05 9.576716e-01 9.665238e-01
77  -0.17243268 0.030821638  2.212158e-08 9.597597e-01 9.597597e-01
78  -0.19861817 0.039611293  5.325998e-07 9.665238e-01 9.665238e-01
79   2.39923606 0.150864650  6.017495e-57           NA           NA
80  -0.65052895 0.103431381  3.185441e-10           NA           NA
81   1.27393827 0.300926352  2.301959e-05           NA           NA
82  20.48354470 0.586874075 6.743251e-267           NA           NA
83   1.10156494 0.279453052  8.085201e-05           NA           NA
84  -0.84432092 0.155532802  5.680691e-08           NA           NA
85   0.11456621 0.022608892  4.034818e-07           NA           NA
86   0.11174010 0.027174149  3.922463e-05           NA           NA
87   2.55171870 0.243573148  1.111305e-25           NA           NA
88  -1.44367601 0.256922194  1.919352e-08           NA           NA
89   2.63813130 0.251250600  8.638013e-26           NA           NA
90  -1.31770917 0.350421416  1.696719e-04           NA           NA
91  -3.84720025 0.577600349  2.725983e-11           NA           NA
92  -3.77821552 0.742726128  3.638552e-07           NA           NA
93  -2.48248806 0.420130448  3.445029e-09           NA           NA
94   2.98118024 0.283921927  8.638013e-26           NA           NA
95   3.27658389 0.312764826  1.111305e-25           NA           NA
96  -1.40348646 0.143073862  1.024509e-22           NA           NA
97  -1.32627199 0.170326027  6.880015e-15           NA           NA
98   2.37002989 0.417723445  1.397607e-08           NA           NA
99   1.26006070 0.206274590  1.004731e-09           NA           NA
100  1.14392760 0.175988862  8.032001e-11           NA           NA
101 -1.84261707 0.299173243  7.319172e-10           NA           NA
102  1.30290616 0.324949745  6.083147e-05           NA           NA
103 -0.10331516 0.024488632  2.454941e-05           NA           NA
104  0.14036286 0.021702950  9.966680e-11           NA           NA
105  1.18411199 0.295322027  6.083147e-05           NA           NA
106  1.25587533 0.329409924  1.375682e-04           NA           NA
107 -4.81819331 0.338817098  6.823380e-46           NA           NA
108 -1.26824285 0.314411143  5.490346e-05           NA           NA
     inst_Q_pval method_Q_pval
1   2.203724e-01    0.39749462
2   1.983898e-07    0.25259259
3   4.312500e-01    0.79424530
4   9.603524e-01            NA
5   2.654757e-01    0.47721182
6   3.226154e-01    0.98524952
7   7.944482e-01    0.94929329
8   3.414803e-01    0.99814065
9   7.658291e-01    0.98465014
10  6.073590e-01            NA
11  8.569471e-01    0.98307615
12  2.129318e-01    0.89616520
13  5.960846e-01            NA
14  3.314384e-01    0.99219730
15  4.096193e-01    0.92963067
16  9.145623e-01            NA
17  8.102286e-01    0.98398107
18  7.078834e-01    0.98760822
19  4.692202e-01    0.99808891
20  4.245707e-01    0.62505460
21  2.640161e-01    0.70420896
22  1.642074e-01    0.93753507
23  3.125807e-01    0.99841565
24  3.052540e-01    0.96972636
25  5.004712e-01    0.65609849
26  5.916775e-01    0.59507823
27  1.959620e-01    0.80585895
28  1.380384e-01    0.97328135
29  4.273608e-01    0.99155427
30  5.619184e-01    0.99822267
31  7.254970e-01    0.97987814
32  4.929829e-01    0.99978987
33  3.932827e-01    0.99788886
34  8.376370e-01    0.99953056
35  2.256333e-01    0.97738358
36  4.122982e-02    0.59105780
37  3.358432e-01    0.99169631
38  8.033788e-01    0.97197293
39  6.552055e-01    0.82263335
40  7.055430e-01    0.89012779
41  4.902574e-02    0.51271336
42  4.009007e-01    0.99658883
43  7.349324e-01            NA
44  4.946887e-02    0.85875202
45  3.577977e-01    0.82965022
46  5.886618e-01    0.22925904
47  2.478625e-11    0.65585989
48  1.920977e-02    0.62837001
49  5.455319e-02    0.99691631
50  2.181672e-01    0.98359736
51  4.291592e-02    0.59310786
52  3.677961e-02            NA
53  3.652151e-01    0.93229010
54  1.545172e-01    0.99125860
55  6.212731e-02    0.98697050
56  3.543796e-03    0.66314796
57  2.248469e-01    0.99152323
58  3.724027e-01    0.24127696
59  5.786149e-01    0.99776350
60  1.513309e-01            NA
61  4.378885e-01    0.87718026
62  3.205239e-02    0.61097216
63  2.201633e-01    0.71769050
64  1.353480e-01    0.18909934
65  5.728204e-07    0.03966958
66  9.988027e-02    0.70318846
67  7.475101e-02    0.62884641
68  1.924860e-01    0.91493274
69  1.822955e-01    0.63763244
70  3.230307e-01    0.99502295
71  4.196328e-02    0.41697513
72  1.828490e-01    0.67442276
73  2.723374e-01    0.60021734
74  1.177621e-03    0.05615641
75  8.744521e-01    0.99623409
76  7.514557e-02    0.91885828
77  2.490760e-01    0.91929667
78  1.553264e-01    0.83332826
79            NA            NA
80            NA            NA
81            NA            NA
82            NA            NA
83            NA            NA
84            NA            NA
85            NA            NA
86            NA            NA
87            NA            NA
88            NA            NA
89            NA            NA
90            NA            NA
91            NA            NA
92            NA            NA
93            NA            NA
94            NA            NA
95            NA            NA
96            NA            NA
97            NA            NA
98            NA            NA
99            NA            NA
100           NA            NA
101           NA            NA
102           NA            NA
103           NA            NA
104           NA            NA
105           NA            NA
106           NA            NA
107           NA            NA
108           NA            NA</code></pre>
</div>
<div class="sourceCode cell-code" id="cb240"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb240-1"><a href="#cb240-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(allres, <span class="st">"forward_mr.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="are-trans-instruments-more-pleiotropic" class="level2">
<h2 class="anchored" data-anchor-id="are-trans-instruments-more-pleiotropic">Are trans instruments more pleiotropic?</h2>
<div class="cell" data-hash="all_cis_mr_cache/html/unnamed-chunk-28_67a8900f3175abe3df35498fdc171920">
<div class="sourceCode cell-code" id="cb241"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb241-1"><a href="#cb241-1" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>fdr.outcome <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(dat<span class="sc">$</span>pval.outcome, <span class="st">"fdr"</span>)</span>
<span id="cb241-2"><a href="#cb241-2" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(dat<span class="sc">$</span>fdr <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
 FALSE   TRUE 
133445   5158 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb243"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb243-1"><a href="#cb243-1" aria-hidden="true" tabindex="-1"></a>dat <span class="sc">%&gt;%</span></span>
<span id="cb243-2"><a href="#cb243-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(SNP, cistrans.exposure) <span class="sc">%&gt;%</span></span>
<span id="cb243-3"><a href="#cb243-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">duplicated</span>(id.outcome)) <span class="sc">%&gt;%</span></span>
<span id="cb243-4"><a href="#cb243-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">n=</span><span class="fu">sum</span>(fdr.outcome <span class="sc">&lt;</span> <span class="fl">0.05</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb243-5"><a href="#cb243-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(cistrans.exposure) <span class="sc">%&gt;%</span></span>
<span id="cb243-6"><a href="#cb243-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">m=</span><span class="fu">mean</span>(n), <span class="at">sd=</span><span class="fu">sd</span>(n), <span class="at">med=</span><span class="fu">median</span>(n), <span class="at">min=</span><span class="fu">min</span>(n), <span class="at">max=</span><span class="fu">max</span>(n))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'SNP'. You can override using the `.groups`
argument.</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 2 × 6
  cistrans.exposure     m    sd   med   min   max
  &lt;chr&gt;             &lt;dbl&gt; &lt;dbl&gt; &lt;int&gt; &lt;int&gt; &lt;int&gt;
1 cis               0.199 0.609     0     0     8
2 trans             0.582 1.09      0     0     9</code></pre>
</div>
</div>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>