<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Gibran Hemani">

<title>Reverse MR analysis of PRS results</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="reverse_mr_files/libs/clipboard/clipboard.min.js"></script>
<script src="reverse_mr_files/libs/quarto-html/quarto.js"></script>
<script src="reverse_mr_files/libs/quarto-html/popper.min.js"></script>
<script src="reverse_mr_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="reverse_mr_files/libs/quarto-html/anchor.min.js"></script>
<link href="reverse_mr_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="reverse_mr_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="reverse_mr_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="reverse_mr_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="reverse_mr_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Reverse MR analysis of PRS results</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Gibran Hemani </p>
          </div>
  </div>
    
  
    
  </div>
  

</header>

<div class="cell" data-hash="reverse_mr_cache/html/unnamed-chunk-1_57da1c944e1935178242a985a1ac4ff3">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(here)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(TwoSampleMR)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyr)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"all.rdata"</span>))</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>prs_pairs <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"prs_pairs.rds"</span>))</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>dat <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"reverse_mr_dat.rds"</span>))</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>idinfo <span class="ot">&lt;-</span> ieugwasr<span class="sc">::</span><span class="fu">gwasinfo</span>(<span class="fu">unique</span>(dat<span class="sc">$</span>id.exposure))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>API: public: http://gwas-api.mrcieu.ac.uk/</code></pre>
</div>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>keep_ids <span class="ot">&lt;-</span> <span class="fu">unique</span>(dat<span class="sc">$</span>id.exposure)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>keep_ids <span class="ot">&lt;-</span> keep_ids[<span class="sc">!</span>keep_ids <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"ieu-a-299"</span>, <span class="st">"ieu-a-300"</span>, <span class="st">"ieu-a-301"</span>, <span class="st">"ieu-a-302"</span>, <span class="st">"ieu-b-38"</span>, <span class="st">"ieu-b-39"</span>)]</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(dat<span class="sc">$</span>id.exposure <span class="sc">%in%</span> keep_ids)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
 TRUE 
89461 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>dat <span class="ot">&lt;-</span> <span class="fu">subset</span>(dat, id.exposure <span class="sc">%in%</span> keep_ids)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>id.exposure[dat<span class="sc">$</span>id.exposure <span class="sc">==</span> <span class="st">"ebi-a-GCST003374"</span>] <span class="ot">&lt;-</span> <span class="st">"ieu-a-1102"</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>id.exposure[dat<span class="sc">$</span>id.exposure <span class="sc">==</span> <span class="st">"ebi-a-GCST005047"</span>] <span class="ot">&lt;-</span> <span class="st">"ieu-a-26"</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>id.exposure[dat<span class="sc">$</span>id.exposure <span class="sc">==</span> <span class="st">"ieu-a-1024"</span>] <span class="ot">&lt;-</span> <span class="st">"ieu-b-18"</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>id.exposure[dat<span class="sc">$</span>id.exposure <span class="sc">==</span> <span class="st">"ieu-a-1187"</span>] <span class="ot">&lt;-</span> <span class="st">"ieu-a-805"</span></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>exposure[dat<span class="sc">$</span>id.exposure <span class="sc">==</span> <span class="st">"ieu-a-805"</span>] <span class="ot">&lt;-</span> <span class="st">"Multiple sclerosis"</span></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>prs_pairs<span class="sc">$</span>opengwasid[prs_pairs<span class="sc">$</span>opengwasid <span class="sc">==</span> <span class="st">"ieu-a-1187"</span>] <span class="ot">&lt;-</span> <span class="st">"ieu-a-805"</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>prs_pairs<span class="sc">$</span>opengwasid[prs_pairs<span class="sc">$</span>opengwasid <span class="sc">==</span> <span class="st">"ieu-a-1024"</span>] <span class="ot">&lt;-</span> <span class="st">"ieu-b-18"</span></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">unique</span>(dat<span class="sc">$</span>id.exposure) <span class="sc">%in%</span> prs_pairs<span class="sc">$</span>opengwasid)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
TRUE 
  21 </code></pre>
</div>
</div>
<p>Perform MR</p>
<div class="cell" data-hash="reverse_mr_cache/html/unnamed-chunk-2_b3fe4c1c64e2e651ac7cbffacabe8bf0">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>rres <span class="ot">&lt;-</span> <span class="fu">suppressMessages</span>(<span class="fu">mr</span>(dat, <span class="at">method=</span><span class="fu">c</span>(<span class="st">"mr_ivw"</span>, <span class="st">"mr_wald_ratio"</span>)))</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>rres<span class="sc">$</span>fdr <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(rres<span class="sc">$</span>pval, <span class="st">"fdr"</span>)</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(rres, <span class="at">file=</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"reverse_mr.rds"</span>))</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(rres<span class="sc">$</span>fdr <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
FALSE  TRUE 
 2057  1588 </code></pre>
</div>
</div>
<p>Compare against PRS</p>
<div class="cell" data-hash="reverse_mr_cache/html/unnamed-chunk-3_600df83eb2f1f156651618805c4f85f5">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>temp <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(prs_pairs, rres, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"prot"</span><span class="ot">=</span><span class="st">"outcome"</span>, <span class="st">"opengwasid"</span><span class="ot">=</span><span class="st">"id.exposure"</span>))</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(temp)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 887  24</code></pre>
</div>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(temp, <span class="fu">aes</span>(<span class="at">x=</span>Estimate, <span class="at">y=</span>b)) <span class="sc">+</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_smooth</span>(<span class="at">method=</span><span class="st">"lm"</span>) <span class="sc">+</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="fu">labs</span>(<span class="at">x=</span><span class="st">"PRS estimate"</span>, <span class="at">y=</span><span class="st">"MR estimate"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`geom_smooth()` using formula = 'y ~ x'</code></pre>
</div>
<div class="cell-output-display">
<p><img src="reverse_mr_files/figure-html/unnamed-chunk-3-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Good agreement between PRS and IVW results</p>
<p>Heterogeneity</p>
<div class="cell" data-hash="reverse_mr_cache/html/unnamed-chunk-4_f920639ad2587bdb1620164d49c45e49">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>het <span class="ot">&lt;-</span> <span class="fu">mr_heterogeneity</span>(dat, <span class="at">method=</span><span class="fu">c</span>(<span class="st">"mr_ivw"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Not enough SNPs available for Heterogeneity analysis of 'ieu-b-89' on 'BTN3A2'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Not enough SNPs available for Heterogeneity analysis of 'ieu-b-89' on 'C2'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Not enough SNPs available for Heterogeneity analysis of 'ieu-b-89' on 'MICB_MICA'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Not enough SNPs available for Heterogeneity analysis of 'ieu-b-89' on 'NPTX1'</code></pre>
</div>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>het<span class="sc">$</span>fdr <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(het<span class="sc">$</span>Q_pval, <span class="st">"fdr"</span>)</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(het<span class="sc">$</span>fdr <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
FALSE  TRUE 
  650  2991 </code></pre>
</div>
</div>
<p>Some SNPs have no heterogeneity</p>
<div class="cell" data-hash="reverse_mr_cache/html/unnamed-chunk-5_d945c3409070571289263a1cb824b410">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>nohet <span class="ot">&lt;-</span> <span class="fu">subset</span>(het, fdr <span class="sc">&gt;</span> <span class="fl">0.05</span>)</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(nohet)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>        id.exposure id.outcome outcome exposure                    method
1  ebi-a-GCST003116      ACAA1   ACAA1      CAD Inverse variance weighted
2  ebi-a-GCST003116       ACE2    ACE2      CAD Inverse variance weighted
6  ebi-a-GCST003116       ADH4    ADH4      CAD Inverse variance weighted
9  ebi-a-GCST003116       AGXT    AGXT      CAD Inverse variance weighted
10 ebi-a-GCST003116    ALDH3A1 ALDH3A1      CAD Inverse variance weighted
16 ebi-a-GCST003116     BAIAP2  BAIAP2      CAD Inverse variance weighted
          Q Q_df     Q_pval        fdr
1  53.94922   41 0.08468799 0.09838831
2  53.26078   41 0.09499853 0.10963222
6  44.01269   41 0.34526117 0.36746446
9  56.06270   41 0.05864017 0.06968304
10 54.79628   41 0.07328239 0.08582219
16 40.31983   41 0.50068812 0.52190251</code></pre>
</div>
</div>
<p>Select SNPs that have no heterogeneity and a significant IVW assoc</p>
<div class="cell" data-hash="reverse_mr_cache/html/unnamed-chunk-6_2989252f9d42d4dc6e9b9c383d734940">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>resnohet <span class="ot">&lt;-</span> <span class="fu">subset</span>(rres, (<span class="fu">paste</span>(exposure, outcome) <span class="sc">%in%</span> <span class="fu">paste</span>(nohet<span class="sc">$</span>exposure, nohet<span class="sc">$</span>outcome) <span class="sc">|</span> nsnp <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">&amp;</span> fdr <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(resnohet<span class="sc">$</span>exposure) <span class="sc">%&gt;%</span> as.data.frame</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                Var1 Freq
1                                CAD    2
2                                CKD  182
3           Diastolic Blood Pressure    1
4                               HDLc    8
5                Lung adenocarcinoma    1
6         Lung cancer (ever smokers)    3
7                                 MS    4
8  Oral cavity and pharyngeal cancer    3
9                      Schizophrenia    1
10         Squamous cell lung cancer    3
11                            Stroke    1
12           Systolic Blood Pressure    3
13                               T2D   33
14                     Triglycerides   28</code></pre>
</div>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(resnohet)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 273  10</code></pre>
</div>
</div>
<p>These assocs seem to be most consistent with being reverse causal</p>
<div class="cell" data-hash="reverse_mr_cache/html/unnamed-chunk-7_a84c7b51e581c83777a98d40eb4bde47">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>resnohet <span class="ot">&lt;-</span> <span class="fu">left_join</span>(resnohet, nohet <span class="sc">%&gt;%</span> <span class="fu">select</span>(id.exposure, id.outcome, Q_pval), <span class="at">by=</span><span class="fu">c</span>(<span class="st">"id.exposure"</span>, <span class="st">"id.outcome"</span>))</span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(resnohet)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>       id.exposure id.outcome outcome exposure                    method nsnp
1 ebi-a-GCST003116      ACAA1   ACAA1      CAD Inverse variance weighted   42
2 ebi-a-GCST003116       TNXB    TNXB      CAD Inverse variance weighted   42
3 ebi-a-GCST003374       ACP6    ACP6      CKD Inverse variance weighted    4
4 ebi-a-GCST003374      ACTA2   ACTA2      CKD Inverse variance weighted    4
5 ebi-a-GCST003374      ADAM8   ADAM8      CKD Inverse variance weighted    4
6 ebi-a-GCST003374     ADGRE2  ADGRE2      CKD Inverse variance weighted    4
            b         se         pval          fdr     Q_pval
1  0.06976545 0.01787638 9.514100e-05 7.362822e-04 0.08468799
2 -0.05814215 0.01679191 5.351619e-04 2.942180e-03 0.05072344
3  0.10814219 0.01972203 4.174190e-08 1.188666e-06 0.91343038
4  0.15081581 0.02744867 3.918923e-08 1.133688e-06 0.96827787
5  0.08376711 0.02870854 3.524574e-03 1.209122e-02 0.66486713
6  0.09477291 0.02830538 8.132962e-04 4.011454e-03 0.73217235</code></pre>
</div>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(resnohet)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 273  11</code></pre>
</div>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(resnohet, <span class="at">file=</span><span class="fu">here</span>(<span class="st">"results"</span>, <span class="st">"resnohet.csv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-hash="reverse_mr_cache/html/unnamed-chunk-8_8d378e17f60880c68c63c0792c42494d">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>resnohetall <span class="ot">&lt;-</span> <span class="fu">subset</span>(rres, (<span class="fu">paste</span>(exposure, outcome) <span class="sc">%in%</span> <span class="fu">paste</span>(nohet<span class="sc">$</span>exposure, nohet<span class="sc">$</span>outcome) <span class="sc">|</span> nsnp <span class="sc">==</span> <span class="dv">1</span>))</span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(resnohetall<span class="sc">$</span>exposure) <span class="sc">%&gt;%</span> as.data.frame</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                Var1 Freq
1                                 AD    2
2                                ALS    3
3                                CAD   25
4                                CKD  244
5           Diastolic Blood Pressure    1
6                 Endometrial cancer   36
7                               HDLc    9
8                Lung adenocarcinoma   34
9         Lung cancer (ever smokers)   74
10                                MS    6
11 Oral cavity and pharyngeal cancer    4
12                    Ovarian cancer    1
13                     Schizophrenia    2
14         Squamous cell lung cancer   56
15                            Stroke   31
16           Systolic Blood Pressure    3
17                               T2D   93
18                     Triglycerides   30</code></pre>
</div>
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(resnohetall)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 654  10</code></pre>
</div>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(resnohetall, <span class="at">file=</span><span class="fu">here</span>(<span class="st">"results"</span>, <span class="st">"resnohetall.csv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Analyse the ones with large heterogeneity</p>
<ol type="1">
<li>Find outliers</li>
<li>Re-estimate with outliers removed</li>
</ol>
<div class="cell" data-hash="reverse_mr_cache/html/unnamed-chunk-9_5a057bc9d45d6aee75d579a741fbdf0c">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>est_qj <span class="ot">&lt;-</span> <span class="cf">function</span>(d) {</span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">nrow</span>(d) <span class="sc">&lt;</span> <span class="dv">2</span>) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>    r <span class="ot">&lt;-</span> <span class="fu">suppressMessages</span>(<span class="fu">mr</span>(d, <span class="at">method=</span><span class="st">"mr_ivw"</span>) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">what=</span><span class="st">"All"</span>))</span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a>    bhat <span class="ot">&lt;-</span> r<span class="sc">$</span>b</span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a>    ss <span class="ot">&lt;-</span> <span class="fu">suppressMessages</span>(<span class="fu">mr_singlesnp</span>(d)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"All"</span>, SNP)) <span class="sc">%&gt;%</span></span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">w =</span> <span class="dv">1</span><span class="sc">/</span>(se<span class="sc">^</span><span class="dv">2</span>), <span class="at">qj =</span> w <span class="sc">*</span> (bhat <span class="sc">-</span> b)<span class="sc">^</span><span class="dv">2</span>, <span class="at">qj_pval =</span> <span class="fu">pchisq</span>(qj, <span class="at">df=</span><span class="dv">1</span>, <span class="at">low=</span>F), <span class="at">qj_fdr =</span> <span class="fu">p.adjust</span>(qj_pval, <span class="st">"fdr"</span>))</span>
<span id="cb36-7"><a href="#cb36-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-8"><a href="#cb36-8" aria-hidden="true" tabindex="-1"></a>    d_noout <span class="ot">&lt;-</span> d <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span>SNP <span class="sc">%in%</span> <span class="fu">subset</span>(ss, <span class="fu">p.adjust</span>(qj_pval, <span class="st">"fdr"</span>) <span class="sc">&lt;</span> <span class="fl">0.05</span>)<span class="sc">$</span>SNP)</span>
<span id="cb36-9"><a href="#cb36-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-10"><a href="#cb36-10" aria-hidden="true" tabindex="-1"></a>    r2 <span class="ot">&lt;-</span> <span class="fu">suppressMessages</span>(<span class="fu">mr</span>(d_noout, <span class="at">method=</span><span class="st">"mr_ivw"</span>)) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">what=</span><span class="st">"Outliers removed"</span>)</span>
<span id="cb36-11"><a href="#cb36-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">list</span>(<span class="at">qj=</span>ss, <span class="at">r=</span><span class="fu">bind_rows</span>(r, r2)))</span>
<span id="cb36-12"><a href="#cb36-12" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb36-13"><a href="#cb36-13" aria-hidden="true" tabindex="-1"></a>qj <span class="ot">&lt;-</span> <span class="fu">group_by</span>(dat, <span class="fu">paste</span>(id.exposure, id.outcome)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-14"><a href="#cb36-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_map</span>(<span class="sc">~</span> <span class="fu">est_qj</span>(.))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in summary.lm(stats::lm(b_out ~ -1 + b_exp, weights = 1/se_out^2)):
essentially perfect fit: summary may be unreliable

Warning in summary.lm(stats::lm(b_out ~ -1 + b_exp, weights = 1/se_out^2)):
essentially perfect fit: summary may be unreliable</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in summary.lm(mod): essentially perfect fit: summary may be unreliable</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get(all_method[i])(beta.exposure, beta.outcome, se.exposure, :
Collinearities in MR Egger, try LD pruning the exposure variables.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in summary.lm(stats::lm(b_out ~ -1 + b_exp, weights = 1/se_out^2)):
essentially perfect fit: summary may be unreliable

Warning in summary.lm(stats::lm(b_out ~ -1 + b_exp, weights = 1/se_out^2)):
essentially perfect fit: summary may be unreliable

Warning in summary.lm(stats::lm(b_out ~ -1 + b_exp, weights = 1/se_out^2)):
essentially perfect fit: summary may be unreliable</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in summary.lm(mod): essentially perfect fit: summary may be unreliable</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get(all_method[i])(beta.exposure, beta.outcome, se.exposure, :
Collinearities in MR Egger, try LD pruning the exposure variables.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in summary.lm(stats::lm(b_out ~ -1 + b_exp, weights = 1/se_out^2)):
essentially perfect fit: summary may be unreliable

Warning in summary.lm(stats::lm(b_out ~ -1 + b_exp, weights = 1/se_out^2)):
essentially perfect fit: summary may be unreliable

Warning in summary.lm(stats::lm(b_out ~ -1 + b_exp, weights = 1/se_out^2)):
essentially perfect fit: summary may be unreliable</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in summary.lm(mod): essentially perfect fit: summary may be unreliable</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get(all_method[i])(beta.exposure, beta.outcome, se.exposure, :
Collinearities in MR Egger, try LD pruning the exposure variables.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in summary.lm(stats::lm(b_out ~ -1 + b_exp, weights = 1/se_out^2)):
essentially perfect fit: summary may be unreliable

Warning in summary.lm(stats::lm(b_out ~ -1 + b_exp, weights = 1/se_out^2)):
essentially perfect fit: summary may be unreliable

Warning in summary.lm(stats::lm(b_out ~ -1 + b_exp, weights = 1/se_out^2)):
essentially perfect fit: summary may be unreliable</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in summary.lm(mod): essentially perfect fit: summary may be unreliable</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get(all_method[i])(beta.exposure, beta.outcome, se.exposure, :
Collinearities in MR Egger, try LD pruning the exposure variables.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in summary.lm(stats::lm(b_out ~ -1 + b_exp, weights = 1/se_out^2)):
essentially perfect fit: summary may be unreliable

Warning in summary.lm(stats::lm(b_out ~ -1 + b_exp, weights = 1/se_out^2)):
essentially perfect fit: summary may be unreliable

Warning in summary.lm(stats::lm(b_out ~ -1 + b_exp, weights = 1/se_out^2)):
essentially perfect fit: summary may be unreliable</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in summary.lm(mod): essentially perfect fit: summary may be unreliable</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get(all_method[i])(beta.exposure, beta.outcome, se.exposure, :
Collinearities in MR Egger, try LD pruning the exposure variables.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in summary.lm(stats::lm(b_out ~ -1 + b_exp, weights = 1/se_out^2)):
essentially perfect fit: summary may be unreliable</code></pre>
</div>
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a>qjs <span class="ot">&lt;-</span> <span class="fu">lapply</span>(qj, \(x) x<span class="sc">$</span>qj) <span class="sc">%&gt;%</span> <span class="fu">bind_rows</span>()</span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a>res_out <span class="ot">&lt;-</span> <span class="fu">lapply</span>(qj, \(x) x<span class="sc">$</span>r) <span class="sc">%&gt;%</span> <span class="fu">bind_rows</span>() <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">names</span>(.)[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>])</span>
<span id="cb53-3"><a href="#cb53-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_out)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>       id.exposure id.outcome outcome exposure                    method nsnp
1 ebi-a-GCST003116      ACAA1   ACAA1      CAD Inverse variance weighted   42
2 ebi-a-GCST003116      ACAA1   ACAA1      CAD Inverse variance weighted   41
3 ebi-a-GCST003116       ACE2    ACE2      CAD Inverse variance weighted   42
4 ebi-a-GCST003116       ACE2    ACE2      CAD Inverse variance weighted   41
5 ebi-a-GCST003116       ACP5    ACP5      CAD Inverse variance weighted   42
6 ebi-a-GCST003116       ACP5    ACP5      CAD Inverse variance weighted   42
              b         se         pval             what
1  0.0697654498 0.01787638 0.0000951410              All
2  0.0623114925 0.01621834 0.0001220162 Outliers removed
3 -0.0042008270 0.01580907 0.7904528075              All
4  0.0004931342 0.01411234 0.9721248046 Outliers removed
5 -0.0110960509 0.01851338 0.5489375253              All
6 -0.0110960509 0.01851338 0.5489375253 Outliers removed</code></pre>
</div>
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(res_out)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 8620   10</code></pre>
</div>
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(qjs, <span class="at">file=</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"reverse_mr_qj.rds"</span>))</span>
<span id="cb57-2"><a href="#cb57-2" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(res_out, <span class="at">file=</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"reverse_mr_outliers.rds"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Results table</p>
<div class="cell" data-hash="reverse_mr_cache/html/unnamed-chunk-10_8d148c72f6d3faefa9c621ec65c011e6">
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a>res_out <span class="sc">%&gt;%</span> <span class="fu">select</span>(exposure, <span class="at">id=</span>id.exposure, outcome, nsnp, b, se, pval, what) <span class="sc">%&gt;%</span> <span class="fu">write.csv</span>(., <span class="at">file=</span><span class="fu">here</span>(<span class="st">"results"</span>, <span class="st">"reverse_mr_results.csv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Example where there is no heterogeneity</p>
<div class="cell" data-hash="reverse_mr_cache/html/unnamed-chunk-11_8dbff4344834f869e80a1b156076813f">
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a>dr <span class="ot">&lt;-</span> <span class="fu">subset</span>(dat, id.exposure<span class="sc">==</span>resnohet<span class="sc">$</span>id.exposure[<span class="dv">1</span>] <span class="sc">&amp;</span> id.outcome<span class="sc">==</span>resnohet<span class="sc">$</span>id.outcome[<span class="dv">1</span>])</span>
<span id="cb59-2"><a href="#cb59-2" aria-hidden="true" tabindex="-1"></a>drr <span class="ot">&lt;-</span> <span class="fu">suppressMessages</span>(<span class="fu">mr</span>(dr))</span>
<span id="cb59-3"><a href="#cb59-3" aria-hidden="true" tabindex="-1"></a><span class="fu">mr_scatter_plot</span>(drr, dr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>$`ebi-a-GCST003116.ACAA1`</code></pre>
</div>
<div class="cell-output-display">
<p><img src="reverse_mr_files/figure-html/unnamed-chunk-11-1.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
attr(,"split_type")
[1] "data.frame"
attr(,"split_labels")
       id.exposure id.outcome
1 ebi-a-GCST003116      ACAA1</code></pre>
</div>
</div>
<p>Example where PRS result is driven by heterogeneity</p>
<div class="cell" data-hash="reverse_mr_cache/html/unnamed-chunk-12_2728286a618606a950f017d3e0ef4ac8">
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a>temp <span class="ot">&lt;-</span> <span class="fu">arrange</span>(het, fdr)</span>
<span id="cb62-2"><a href="#cb62-2" aria-hidden="true" tabindex="-1"></a>dr2 <span class="ot">&lt;-</span> <span class="fu">subset</span>(dat, id.exposure<span class="sc">==</span>temp<span class="sc">$</span>id.exposure[<span class="dv">1</span>] <span class="sc">&amp;</span> id.outcome<span class="sc">==</span>temp<span class="sc">$</span>id.outcome[<span class="dv">1</span>])</span>
<span id="cb62-3"><a href="#cb62-3" aria-hidden="true" tabindex="-1"></a>drr2 <span class="ot">&lt;-</span> <span class="fu">suppressMessages</span>(<span class="fu">mr</span>(dr2))</span>
<span id="cb62-4"><a href="#cb62-4" aria-hidden="true" tabindex="-1"></a><span class="fu">mr_scatter_plot</span>(drr2, dr2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>$`ebi-a-GCST003116.GRN`</code></pre>
</div>
<div class="cell-output-display">
<p><img src="reverse_mr_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
attr(,"split_type")
[1] "data.frame"
attr(,"split_labels")
       id.exposure id.outcome
1 ebi-a-GCST003116        GRN</code></pre>
</div>
</div>
<p>Outlier analysis. How many have outliers but after removing them the original assoc still remains</p>
<div class="cell" data-hash="reverse_mr_cache/html/unnamed-chunk-13_12213d1062ba9166abe45f4cb311fc2c">
<div class="sourceCode cell-code" id="cb65"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a>thresh <span class="ot">&lt;-</span> <span class="fu">subset</span>(rres, fdr <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(pval)) <span class="sc">%&gt;%</span> {<span class="fu">max</span>(.<span class="sc">$</span>pval)}</span>
<span id="cb65-2"><a href="#cb65-2" aria-hidden="true" tabindex="-1"></a>res_out_sum <span class="ot">&lt;-</span> res_out <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(id.exposure, id.outcome) <span class="sc">%&gt;%</span></span>
<span id="cb65-3"><a href="#cb65-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(</span>
<span id="cb65-4"><a href="#cb65-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">nsnp_orig =</span> nsnp[<span class="dv">1</span>],</span>
<span id="cb65-5"><a href="#cb65-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">noutliers =</span> nsnp[<span class="dv">1</span>] <span class="sc">-</span> nsnp[<span class="dv">2</span>],</span>
<span id="cb65-6"><a href="#cb65-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">fdr_orig =</span> pval[<span class="dv">1</span>] <span class="sc">&lt;</span> thresh,</span>
<span id="cb65-7"><a href="#cb65-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">fdr_out =</span> pval[<span class="dv">2</span>] <span class="sc">&lt;</span> thresh</span>
<span id="cb65-8"><a href="#cb65-8" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(fdr_orig)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'id.exposure'. You can override using the
`.groups` argument.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb67"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(res_out_sum<span class="sc">$</span>fdr_orig <span class="sc">&amp;</span> res_out_sum<span class="sc">$</span>fdr_out, <span class="at">na.rm=</span>T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 423</code></pre>
</div>
</div>
<p>Reverse MR summary</p>
<div class="cell" data-hash="reverse_mr_cache/html/unnamed-chunk-14_a359014847d49203b28c1e2f3db167ce">
<div class="sourceCode cell-code" id="cb69"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a><span class="fu">tribble</span>(</span>
<span id="cb69-2"><a href="#cb69-2" aria-hidden="true" tabindex="-1"></a>    <span class="sc">~</span>what, <span class="sc">~</span>n,</span>
<span id="cb69-3"><a href="#cb69-3" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Number of PRS results"</span>, <span class="fu">nrow</span>(prs_pairs),</span>
<span id="cb69-4"><a href="#cb69-4" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Number of reverse MR assocs"</span>, <span class="fu">nrow</span>(rres),</span>
<span id="cb69-5"><a href="#cb69-5" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Reverse MR FDR &lt; 0.05"</span>, <span class="fu">sum</span>(rres<span class="sc">$</span>fdr <span class="sc">&lt;</span> <span class="fl">0.05</span>, <span class="at">na.rm=</span>T),</span>
<span id="cb69-6"><a href="#cb69-6" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Reverse MR FDR &lt; 0.05 and no heterogeneity"</span>, <span class="fu">nrow</span>(resnohet),</span>
<span id="cb69-7"><a href="#cb69-7" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Reverse MR FDR &lt; 0.05 after outlier removal"</span>, <span class="fu">sum</span>(res_out_sum<span class="sc">$</span>fdr_orig <span class="sc">&amp;</span> res_out_sum<span class="sc">$</span>fdr_out, <span class="at">na.rm=</span>T),</span>
<span id="cb69-8"><a href="#cb69-8" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Mean number of instruments"</span>, <span class="fu">mean</span>(res_out_sum<span class="sc">$</span>nsnp_orig),</span>
<span id="cb69-9"><a href="#cb69-9" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Mean number of outliers"</span>, <span class="fu">mean</span>(res_out_sum<span class="sc">$</span>noutliers, <span class="at">na.rm=</span>T)</span>
<span id="cb69-10"><a href="#cb69-10" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 7 Ã— 2
  what                                                n
  &lt;chr&gt;                                           &lt;dbl&gt;
1 Number of PRS results                       30849    
2 Number of reverse MR assocs                  3645    
3 Reverse MR FDR &lt; 0.05                        1588    
4 Reverse MR FDR &lt; 0.05 and no heterogeneity    273    
5 Reverse MR FDR &lt; 0.05 after outlier removal   423    
6 Mean number of instruments                     14.3  
7 Mean number of outliers                         0.902</code></pre>
</div>
</div>
<p>Compare PRS and reverse MR</p>
<div class="cell" data-hash="reverse_mr_cache/html/unnamed-chunk-15_070171c946a99508565c29697a556f21">
<div class="sourceCode cell-code" id="cb71"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a>revmrpairs <span class="ot">&lt;-</span> <span class="fu">subset</span>(res_out_sum, fdr_orig <span class="sc">&amp;</span> fdr_out)</span>
<span id="cb71-2"><a href="#cb71-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(revmrpairs)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 423   6</code></pre>
</div>
<div class="sourceCode cell-code" id="cb73"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a>rmp <span class="ot">&lt;-</span> <span class="fu">subset</span>(res_out, what<span class="sc">==</span><span class="st">"Outliers removed"</span> <span class="sc">&amp;</span> <span class="fu">paste</span>(id.exposure, id.outcome) <span class="sc">%in%</span> <span class="fu">paste</span>(revmrpairs<span class="sc">$</span>id.exposure, revmrpairs<span class="sc">$</span>id.outcome))</span>
<span id="cb73-2"><a href="#cb73-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(rmp)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>         id.exposure id.outcome outcome exposure                    method nsnp
2   ebi-a-GCST003116      ACAA1   ACAA1      CAD Inverse variance weighted   41
177 ebi-a-GCST003116      CNTN4   CNTN4      CAD Inverse variance weighted   42
227 ebi-a-GCST003116     CXCL13  CXCL13      CAD Inverse variance weighted   37
267 ebi-a-GCST003116      ERBB4   ERBB4      CAD Inverse variance weighted   41
347 ebi-a-GCST003116      HMOX2   HMOX2      CAD Inverse variance weighted   41
415 ebi-a-GCST003116      ITGAV   ITGAV      CAD Inverse variance weighted   40
              b         se         pval             what
2    0.06231149 0.01621834 1.220162e-04 Outliers removed
177 -0.07001910 0.01769095 7.561032e-05 Outliers removed
227  0.04577365 0.01816049 1.171856e-02 Outliers removed
267 -0.06034025 0.01625610 2.057472e-04 Outliers removed
347  0.04983681 0.01569445 1.496086e-03 Outliers removed
415 -0.04767240 0.01477194 1.249960e-03 Outliers removed</code></pre>
</div>
<div class="sourceCode cell-code" id="cb75"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb75-1"><a href="#cb75-1" aria-hidden="true" tabindex="-1"></a>temp <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(prs_pairs, rres, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"prot"</span><span class="ot">=</span><span class="st">"outcome"</span>, <span class="st">"opengwasid"</span><span class="ot">=</span><span class="st">"id.exposure"</span>))</span>
<span id="cb75-2"><a href="#cb75-2" aria-hidden="true" tabindex="-1"></a>temp<span class="sc">$</span>repl <span class="ot">&lt;-</span> <span class="fu">paste</span>(temp<span class="sc">$</span>prot, temp<span class="sc">$</span>opengwasid) <span class="sc">%in%</span> <span class="fu">paste</span>(rmp<span class="sc">$</span>id.outcome, rmp<span class="sc">$</span>id.exposure)</span>
<span id="cb75-3"><a href="#cb75-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb75-4"><a href="#cb75-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(temp)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>           UKBPPP_ProteinID   prot     v1       v2 v3    Estimate   Std_Error
1  TREM2:Q9NZC2:OID20731:v1  TREM2 Q9NZC2 OID20731 v1 -0.03292007 0.006103260
2   GBP4:Q96PP9:OID20845:v1   GBP4 Q96PP9 OID20845 v1 -0.03330563 0.006622041
3   CD33:P20138:OID21472:v1   CD33 P20138 OID21472 v1  0.03249780 0.006587509
4   C1QA:P02745:OID20654:v1   C1QA P02745 OID20654 v1 -0.03143641 0.006439390
5  NDRG1:Q92597:OID20792:v1  NDRG1 Q92597 OID20792 v1 -0.03097661 0.006631114
6 BTN3A2:P78410:OID20540:v1 BTN3A2 P78410 OID20540 v1  0.02971657 0.006401337
    t.value      P_value  FDR_P_value
1 -5.393851 6.942740e-08 2.560135e-05
2 -5.029511 4.943037e-07 1.458196e-04
3  4.933246 8.125707e-07 1.997570e-04
4 -4.881894 1.055486e-06 2.224061e-04
5 -4.671405 3.003241e-06 5.537226e-04
6  4.642245 3.459423e-06 5.669609e-04
                             olink_target_fullname        Disease_Name
1 Triggering receptor expressed on myeloid cells 2 Alzheimerâ€™s disease
2                      Guanylate-binding protein 4 Alzheimerâ€™s disease
3                Myeloid cell surface antigen CD33 Alzheimerâ€™s disease
4            Complement C1q subcomponent subunit A Alzheimerâ€™s disease
5                                    Protein NDRG1 Alzheimerâ€™s disease
6               Butyrophilin subfamily 3 member A2 Alzheimerâ€™s disease
  Disease_Code opengwasid        fdr.x  sig id.outcome exposure
1           AD    ieu-b-2 2.999672e-06 TRUE      TREM2       AD
2           AD    ieu-b-2 1.746710e-05 TRUE       GBP4       AD
3           AD    ieu-b-2 2.748574e-05 TRUE       CD33       AD
4           AD    ieu-b-2 3.456550e-05 TRUE       C1QA       AD
5           AD    ieu-b-2 8.691087e-05 TRUE      NDRG1       AD
6           AD    ieu-b-2 9.813116e-05 TRUE     BTN3A2       AD
                     method nsnp            b          se       pval     fdr.y
1 Inverse variance weighted   21 -0.088406431 0.109075673 0.41764971 0.4920380
2 Inverse variance weighted   21 -0.006771840 0.012645614 0.59229796 0.6528352
3 Inverse variance weighted   21 -0.005146903 0.008974738 0.56631440 0.6302950
4 Inverse variance weighted   21 -0.022607576 0.012182481 0.06348993 0.1149061
5 Inverse variance weighted   21 -0.018313266 0.016294366 0.26105471 0.3418089
6 Inverse variance weighted   21  0.008033899 0.022048394 0.71557749 0.7604315
   repl
1 FALSE
2 FALSE
3 FALSE
4 FALSE
5 FALSE
6 FALSE</code></pre>
</div>
<div class="sourceCode cell-code" id="cb77"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb77-1"><a href="#cb77-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(temp<span class="sc">$</span>repl)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
FALSE  TRUE 
  859    28 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb79"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a>temp <span class="sc">%&gt;%</span> <span class="fu">filter</span>(b <span class="sc">&lt;</span> <span class="dv">1</span> <span class="sc">&amp;</span> P_value <span class="sc">&lt;</span> <span class="fl">0.05</span><span class="sc">/</span><span class="fu">nrow</span>(temp)) <span class="sc">%&gt;%</span></span>
<span id="cb79-2"><a href="#cb79-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(repl) <span class="sc">%&gt;%</span></span>
<span id="cb79-3"><a href="#cb79-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(., <span class="fu">aes</span>(<span class="at">x=</span>Estimate, <span class="at">y=</span>b)) <span class="sc">+</span></span>
<span id="cb79-4"><a href="#cb79-4" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb79-5"><a href="#cb79-5" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_smooth</span>(<span class="at">method=</span><span class="st">"lm"</span>) <span class="sc">+</span></span>
<span id="cb79-6"><a href="#cb79-6" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_abline</span>(<span class="at">slope=</span><span class="dv">1</span>, <span class="at">intercept=</span><span class="dv">0</span>, <span class="at">linetype=</span><span class="st">"dotted"</span>) <span class="sc">+</span></span>
<span id="cb79-7"><a href="#cb79-7" aria-hidden="true" tabindex="-1"></a><span class="fu">labs</span>(<span class="at">x=</span><span class="st">"PRS estimate"</span>, <span class="at">y=</span><span class="st">"MR estimate"</span>, <span class="at">colour=</span><span class="st">"MR FDR &lt; 0.05"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`geom_smooth()` using formula = 'y ~ x'</code></pre>
</div>
<div class="cell-output-display">
<p><img src="reverse_mr_files/figure-html/unnamed-chunk-15-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb81"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb81-1"><a href="#cb81-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="fu">here</span>(<span class="st">"images"</span>, <span class="st">"mr_prs_comparison.pdf"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image
`geom_smooth()` using formula = 'y ~ x'</code></pre>
</div>
</div>
<div class="cell" data-hash="reverse_mr_cache/html/unnamed-chunk-16_98495e39ae7df12137d5d3e19c6b8a57">
<div class="sourceCode cell-code" id="cb83"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb83-1"><a href="#cb83-1" aria-hidden="true" tabindex="-1"></a><span class="fu">subset</span>(temp, Estimate <span class="sc">&lt;</span> <span class="dv">0</span> <span class="sc">&amp;</span> b <span class="sc">&gt;</span> <span class="dv">0</span> <span class="sc">&amp;</span> repl)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] UKBPPP_ProteinID      prot                  v1                   
 [4] v2                    v3                    Estimate             
 [7] Std_Error             t.value               P_value              
[10] FDR_P_value           olink_target_fullname Disease_Name         
[13] Disease_Code          opengwasid            fdr.x                
[16] sig                   id.outcome            exposure             
[19] method                nsnp                  b                    
[22] se                    pval                  fdr.y                
[25] repl                 
&lt;0 rows&gt; (or 0-length row.names)</code></pre>
</div>
</div>
<div class="cell" data-hash="reverse_mr_cache/html/unnamed-chunk-17_ee61ee32efc04719462b9d02376d329c">
<div class="sourceCode cell-code" id="cb85"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb85-1"><a href="#cb85-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(rmp) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb85-2"><a href="#cb85-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(rmp, <span class="at">file=</span><span class="fu">here</span>(<span class="st">"results"</span>, <span class="st">"resnohet.csv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Other Summary</p>
<div class="cell" data-hash="reverse_mr_cache/html/unnamed-chunk-18_4e4255ed824cf7bc09b2bf5c34f608be">
<div class="sourceCode cell-code" id="cb86"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a>cis_prs <span class="ot">&lt;-</span> <span class="fu">subset</span>(res_cis, <span class="fu">paste</span>(id.exposure, id.outcome) <span class="sc">%in%</span> <span class="fu">paste</span>(prs_pairs<span class="sc">$</span>prot, prs_pairs<span class="sc">$</span>opengwasid))</span>
<span id="cb86-2"><a href="#cb86-2" aria-hidden="true" tabindex="-1"></a>cis_prs<span class="sc">$</span>fdr <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(cis_prs<span class="sc">$</span>pval, <span class="st">"fdr"</span>)</span>
<span id="cb86-3"><a href="#cb86-3" aria-hidden="true" tabindex="-1"></a>a <span class="ot">&lt;-</span> <span class="fu">subset</span>(cis_prs, fdr <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">code =</span> <span class="fu">paste</span>(id.exposure, id.outcome))</span>
<span id="cb86-4"><a href="#cb86-4" aria-hidden="true" tabindex="-1"></a>res_cis<span class="sc">$</span>fdr <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(res_cis<span class="sc">$</span>pval, <span class="st">"fdr"</span>)</span>
<span id="cb86-5"><a href="#cb86-5" aria-hidden="true" tabindex="-1"></a>b <span class="ot">&lt;-</span> <span class="fu">subset</span>(res_cis, fdr <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">code =</span> <span class="fu">paste</span>(id.exposure, id.outcome))</span>
<span id="cb86-6"><a href="#cb86-6" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(a<span class="sc">$</span>code <span class="sc">%in%</span> b<span class="sc">$</span>code)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 94</code></pre>
</div>
<div class="sourceCode cell-code" id="cb88"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a>colocres <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"cis_coloc_res.rds"</span>))</span>
<span id="cb88-2"><a href="#cb88-2" aria-hidden="true" tabindex="-1"></a>res_cis<span class="sc">$</span>fdr <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(res_cis<span class="sc">$</span>pval, <span class="st">"fdr"</span>)</span>
<span id="cb88-3"><a href="#cb88-3" aria-hidden="true" tabindex="-1"></a><span class="fu">tribble</span>(</span>
<span id="cb88-4"><a href="#cb88-4" aria-hidden="true" tabindex="-1"></a>    <span class="sc">~</span>what, <span class="sc">~</span>n,</span>
<span id="cb88-5"><a href="#cb88-5" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Forward MR tests"</span>, <span class="fu">nrow</span>(res),</span>
<span id="cb88-6"><a href="#cb88-6" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Forward MR tests using cis-only instruments"</span>, <span class="fu">nrow</span>(res_cis),</span>
<span id="cb88-7"><a href="#cb88-7" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Forward MR tests using trans-only instruments"</span>, <span class="fu">nrow</span>(res_trans),</span>
<span id="cb88-8"><a href="#cb88-8" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Cis MR with FDR &lt; 0.05"</span>, <span class="fu">sum</span>(res_cis<span class="sc">$</span>fdr <span class="sc">&lt;</span> <span class="fl">0.05</span>),</span>
<span id="cb88-9"><a href="#cb88-9" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Total tested - reverse MR filter"</span>, <span class="fu">nrow</span>(cis_prs),</span>
<span id="cb88-10"><a href="#cb88-10" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Total discovered - reverse MR filter"</span>, <span class="fu">sum</span>(cis_prs<span class="sc">$</span>fdr <span class="sc">&lt;</span> <span class="fl">0.05</span>),</span>
<span id="cb88-11"><a href="#cb88-11" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Discovery overlap"</span>, <span class="fu">sum</span>(a<span class="sc">$</span>code <span class="sc">%in%</span> b<span class="sc">$</span>code),</span>
<span id="cb88-12"><a href="#cb88-12" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Cis Colocalisation with H4 &gt; 0.8"</span>, <span class="fu">sum</span>(colocres<span class="sc">$</span>PP.H4.abf <span class="sc">&gt;</span> <span class="fl">0.8</span>),</span>
<span id="cb88-13"><a href="#cb88-13" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 8 Ã— 2
  what                                              n
  &lt;chr&gt;                                         &lt;int&gt;
1 Forward MR tests                              22434
2 Forward MR tests using cis-only instruments   15246
3 Forward MR tests using trans-only instruments 21468
4 Cis MR with FDR &lt; 0.05                           94
5 Total tested - reverse MR filter              15246
6 Total discovered - reverse MR filter             94
7 Discovery overlap                                94
8 Cis Colocalisation with H4 &gt; 0.8                 45</code></pre>
</div>
</div>
<p>How many prs_pairs per trait</p>
<div class="cell" data-hash="reverse_mr_cache/html/unnamed-chunk-19_e405553f46c2b0987236acb9a731560d">
<div class="sourceCode cell-code" id="cb90"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(prs_pairs<span class="sc">$</span>opengwasid) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(Freq) <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(., traits <span class="sc">%&gt;%</span> <span class="fu">select</span>(code, opengwasid), <span class="at">by=</span><span class="fu">c</span>(<span class="st">"Var1"</span><span class="ot">=</span><span class="st">"opengwasid"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>               Var1 Freq
1  ebi-a-GCST003116 1469
2  ebi-a-GCST005647 1469
3  ebi-a-GCST005838 1469
4  ebi-a-GCST006464 1469
5        ieu-a-1102 1469
6        ieu-a-1120 1469
7        ieu-a-1121 1469
8        ieu-a-1126 1469
9        ieu-a-1127 1469
10       ieu-a-1228 1469
11         ieu-a-26 1469
12        ieu-a-805 1469
13        ieu-a-984 1469
14        ieu-a-985 1469
15        ieu-a-989 1469
16         ieu-b-18 1469
17          ieu-b-2 1469
18         ieu-b-42 1469
19          ieu-b-7 1469
20         ieu-b-85 1469
21         ieu-b-89 1469
                                                                code
1                                                                CAD
2                                                                ALS
3                                                             Stroke
4                                                 Endometrial cancer
5                                                                CKD
6                                                     Ovarian cancer
7                                   High grade serous ovarian cancer
8                                                      Breast cancer
9  ER+ Breast cancer (Combined Oncoarray; iCOGS; GWAS meta analysis)
10                    High grade and low grade serous ovarian cancer
11                                                               T2D
12                                                               MDD
13                                               Lung adenocarcinoma
14                                        Lung cancer (ever smokers)
15                                         Squamous cell lung cancer
16                                                                MS
17                                                                AD
18                                                     Schizophrenia
19                                                                PD
20                                                   Prostate cancer
21                                 Oral cavity and pharyngeal cancer</code></pre>
</div>
</div>
<p>Examine CKD - it has a lot of assocs</p>
<div class="cell" data-hash="reverse_mr_cache/html/unnamed-chunk-20_f04d468c95156585481c4805b9df4343">
<div class="sourceCode cell-code" id="cb92"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1" aria-hidden="true" tabindex="-1"></a>ckd <span class="ot">&lt;-</span> <span class="fu">subset</span>(dat, exposure <span class="sc">==</span> <span class="st">"CKD"</span>)</span>
<span id="cb92-2"><a href="#cb92-2" aria-hidden="true" tabindex="-1"></a>ckd <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(id.outcome) <span class="sc">%&gt;%</span></span>
<span id="cb92-3"><a href="#cb92-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(pval.outcome) <span class="sc">%&gt;%</span></span>
<span id="cb92-4"><a href="#cb92-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">slice_head</span>(<span class="at">n=</span><span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb92-5"><a href="#cb92-5" aria-hidden="true" tabindex="-1"></a>    {<span class="fu">table</span>(.<span class="sc">$</span>SNP)}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
rs13333226  rs2453533  rs3812035  rs7805747 
       438         33         56         79 </code></pre>
</div>
</div>
<p>Most of the time the best SNP is a single SNP. Heterogeneity?</p>
<div class="cell" data-hash="reverse_mr_cache/html/unnamed-chunk-21_7b0c92947e9c12b6cda86a48e0da658e">
<div class="sourceCode cell-code" id="cb94"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a>qjs <span class="sc">%&gt;%</span> <span class="fu">filter</span>(exposure <span class="sc">==</span> <span class="st">"CKD"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb94-2"><a href="#cb94-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(SNP) <span class="sc">%&gt;%</span></span>
<span id="cb94-3"><a href="#cb94-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="fu">mean</span>(qj))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 4 Ã— 2
  SNP        `mean(qj)`
  &lt;chr&gt;           &lt;dbl&gt;
1 rs13333226       9.14
2 rs2453533        5.20
3 rs3812035        2.24
4 rs7805747        1.77</code></pre>
</div>
</div>
<p>Only 4 exposure SNPs..</p>
<div class="cell" data-hash="reverse_mr_cache/html/unnamed-chunk-22_11ea759f98581bc617189f74233207eb">
<div class="sourceCode cell-code" id="cb96"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb96-1"><a href="#cb96-1" aria-hidden="true" tabindex="-1"></a><span class="fu">subset</span>(ckd, outcome <span class="sc">==</span> outcome[<span class="dv">1</span>]) <span class="sc">%&gt;%</span> <span class="fu">select</span>(SNP, pval.exposure)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>             SNP pval.exposure
49291 rs13333226   3.80014e-26
49292  rs2453533   5.40008e-12
49293  rs3812035   2.90001e-09
49294  rs7805747   2.09991e-14</code></pre>
</div>
</div>
<p>Plot an example</p>
<div class="cell" data-hash="reverse_mr_cache/html/unnamed-chunk-23_127ba53ff300e9cf1706409fc862b22b">
<div class="sourceCode cell-code" id="cb98"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb98-1"><a href="#cb98-1" aria-hidden="true" tabindex="-1"></a><span class="co"># note that prs_pairs uses different opengwasid</span></span>
<span id="cb98-2"><a href="#cb98-2" aria-hidden="true" tabindex="-1"></a>ckd_top <span class="ot">&lt;-</span> <span class="fu">subset</span>(ckd, outcome <span class="sc">%in%</span> <span class="fu">subset</span>(prs_pairs, opengwasid <span class="sc">==</span> <span class="st">"ieu-a-1102"</span>)<span class="sc">$</span>prot[<span class="dv">1</span><span class="sc">:</span><span class="dv">28</span>])</span>
<span id="cb98-3"><a href="#cb98-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb98-4"><a href="#cb98-4" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(ckd_top, <span class="fu">aes</span>(<span class="at">x=</span>beta.exposure, <span class="at">y=</span>beta.outcome)) <span class="sc">+</span></span>
<span id="cb98-5"><a href="#cb98-5" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb98-6"><a href="#cb98-6" aria-hidden="true" tabindex="-1"></a><span class="fu">facet_wrap</span>(<span class="sc">~</span>id.outcome) <span class="sc">+</span></span>
<span id="cb98-7"><a href="#cb98-7" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_smooth</span>(<span class="at">method=</span><span class="st">"lm"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`geom_smooth()` using formula = 'y ~ x'</code></pre>
</div>
<div class="cell-output-display">
<p><img src="reverse_mr_files/figure-html/unnamed-chunk-23-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>The pattern of association is identical across all these proteins. I donâ€™t know what to make of this. The 4 SNPs for CKD have a consistent effect on loads of blood protein levels?</p>
<p>Look at lung cancer too?</p>
<div class="cell" data-hash="reverse_mr_cache/html/unnamed-chunk-24_abef6a0efdb651f24725b0efd22e4565">
<div class="sourceCode cell-code" id="cb100"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb100-1"><a href="#cb100-1" aria-hidden="true" tabindex="-1"></a>sclc <span class="ot">&lt;-</span> <span class="fu">subset</span>(dat, id.exposure <span class="sc">==</span> <span class="st">"ieu-a-989"</span> <span class="sc">&amp;</span> id.outcome <span class="sc">%in%</span> <span class="fu">subset</span>(res_out_sum, id.exposure<span class="sc">==</span><span class="st">"ieu-a-989"</span>)<span class="sc">$</span>id.outcome)</span>
<span id="cb100-2"><a href="#cb100-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-3"><a href="#cb100-3" aria-hidden="true" tabindex="-1"></a>sclc_top <span class="ot">&lt;-</span> <span class="fu">subset</span>(dat, id.exposure <span class="sc">==</span> <span class="st">"ieu-a-989"</span> <span class="sc">&amp;</span> id.outcome <span class="sc">%in%</span> <span class="fu">subset</span>(prs_pairs, opengwasid<span class="sc">==</span><span class="st">"ieu-a-989"</span>)<span class="sc">$</span>prot[<span class="dv">1</span><span class="sc">:</span><span class="dv">28</span>])</span>
<span id="cb100-4"><a href="#cb100-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-5"><a href="#cb100-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(sclc, <span class="fu">aes</span>(<span class="at">x=</span>beta.exposure, <span class="at">y=</span>beta.outcome)) <span class="sc">+</span></span>
<span id="cb100-6"><a href="#cb100-6" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb100-7"><a href="#cb100-7" aria-hidden="true" tabindex="-1"></a><span class="fu">facet_wrap</span>(<span class="sc">~</span>id.outcome) <span class="sc">+</span></span>
<span id="cb100-8"><a href="#cb100-8" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_smooth</span>(<span class="at">method=</span><span class="st">"lm"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`geom_smooth()` using formula = 'y ~ x'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in qt((1 - level)/2, df): NaNs produced</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in max(ids, na.rm = TRUE): no non-missing arguments to max; returning
-Inf</code></pre>
</div>
<div class="cell-output-display">
<p><img src="reverse_mr_files/figure-html/unnamed-chunk-24-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Plot MR against PRS</p>
<div class="cell" data-hash="reverse_mr_cache/html/unnamed-chunk-25_5f0acfa01c86edf83de6c2f14ea58034">
<div class="sourceCode cell-code" id="cb104"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb104-1"><a href="#cb104-1" aria-hidden="true" tabindex="-1"></a>qjs <span class="sc">%&gt;%</span> <span class="fu">filter</span>(id.exposure<span class="sc">==</span><span class="st">"ieu-a-989"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb104-2"><a href="#cb104-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(SNP) <span class="sc">%&gt;%</span></span>
<span id="cb104-3"><a href="#cb104-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="fu">mean</span>(qj))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 9 Ã— 2
  SNP        `mean(qj)`
  &lt;chr&gt;           &lt;dbl&gt;
1 rs11571815       2.15
2 rs1333040        2.32
3 rs2066295        4.69
4 rs467095         3.93
5 rs501942        56.2 
6 rs56113850       3.03
7 rs7953330        2.02
8 rs8040868        4.94
9 rs9273429       19.6 </code></pre>
</div>
</div>
<div class="cell" data-hash="reverse_mr_cache/html/unnamed-chunk-26_a3b974657b3a3801e321bea6bc2e5405">
<div class="sourceCode cell-code" id="cb106"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb106-1"><a href="#cb106-1" aria-hidden="true" tabindex="-1"></a>res_out_sum <span class="sc">%&gt;%</span></span>
<span id="cb106-2"><a href="#cb106-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(id.exposure) <span class="sc">%&gt;%</span></span>
<span id="cb106-3"><a href="#cb106-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">fdr_out=</span><span class="fu">sum</span>(fdr_out, <span class="at">na.rm=</span>T), <span class="at">fdr_orig=</span><span class="fu">sum</span>(fdr_orig, <span class="at">na.rm=</span>T)) <span class="sc">%&gt;%</span></span>
<span id="cb106-4"><a href="#cb106-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(., traits <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="at">id.exposure=</span>opengwasid, code)) <span class="sc">%&gt;%</span> </span>
<span id="cb106-5"><a href="#cb106-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(., prs_pairs <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(<span class="at">id.exposure=</span>opengwasid) <span class="sc">%&gt;%</span> <span class="fu">summarise</span>(<span class="at">nprs=</span><span class="fu">n</span>())) <span class="sc">%&gt;%</span></span>
<span id="cb106-6"><a href="#cb106-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="fu">desc</span>(nprs)) <span class="sc">%&gt;%</span></span>
<span id="cb106-7"><a href="#cb106-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(code, id.exposure, <span class="at">n_prs=</span>nprs, <span class="at">n_mr=</span>fdr_orig, <span class="at">n_mr_out=</span>fdr_out)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining with `by = join_by(id.exposure)`
Joining with `by = join_by(id.exposure)`</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 11 Ã— 5
   code                       id.exposure      n_prs  n_mr n_mr_out
   &lt;chr&gt;                      &lt;chr&gt;            &lt;int&gt; &lt;int&gt;    &lt;int&gt;
 1 CAD                        ebi-a-GCST003116  1469    18       13
 2 Stroke                     ebi-a-GCST005838  1469     1        1
 3 CKD                        ieu-a-1102        1469   309      308
 4 Ovarian cancer             ieu-a-1120        1469     1        1
 5 T2D                        ieu-a-26          1469    65       58
 6 Lung adenocarcinoma        ieu-a-984         1469     1        1
 7 Lung cancer (ever smokers) ieu-a-985         1469     4        4
 8 Squamous cell lung cancer  ieu-a-989         1469     5        4
 9 MS                         ieu-b-18          1469    53       29
10 Schizophrenia              ieu-b-42          1469    10        4
11 Prostate cancer            ieu-b-85          1469     2        0</code></pre>
</div>
</div>
<p>T2D</p>
<div class="cell" data-hash="reverse_mr_cache/html/unnamed-chunk-27_8281e6a2efbbcb2cbe7b07d0fd2c39d0">
<div class="sourceCode cell-code" id="cb109"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb109-1"><a href="#cb109-1" aria-hidden="true" tabindex="-1"></a><span class="co"># note that prs_pairs uses different opengwasid</span></span>
<span id="cb109-2"><a href="#cb109-2" aria-hidden="true" tabindex="-1"></a>t2d_top <span class="ot">&lt;-</span> <span class="fu">subset</span>(dat, id.exposure <span class="sc">==</span> <span class="st">"ieu-a-26"</span> <span class="sc">&amp;</span> id.outcome <span class="sc">%in%</span> <span class="fu">unique</span>(<span class="fu">subset</span>(prs_pairs, opengwasid<span class="sc">==</span><span class="st">"ieu-a-26"</span>))<span class="sc">$</span>prot[<span class="dv">1</span><span class="sc">:</span><span class="dv">56</span>])</span>
<span id="cb109-3"><a href="#cb109-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb109-4"><a href="#cb109-4" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(t2d_top, <span class="fu">aes</span>(<span class="at">x=</span>beta.exposure, <span class="at">y=</span>beta.outcome)) <span class="sc">+</span></span>
<span id="cb109-5"><a href="#cb109-5" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb109-6"><a href="#cb109-6" aria-hidden="true" tabindex="-1"></a><span class="fu">facet_wrap</span>(<span class="sc">~</span>id.outcome) <span class="sc">+</span></span>
<span id="cb109-7"><a href="#cb109-7" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_smooth</span>(<span class="at">method=</span><span class="st">"lm"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`geom_smooth()` using formula = 'y ~ x'</code></pre>
</div>
<div class="cell-output-display">
<p><img src="reverse_mr_files/figure-html/unnamed-chunk-27-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>MS</p>
<div class="cell" data-hash="reverse_mr_cache/html/unnamed-chunk-28_8deeb60ad9a9230ab869e4b4b34ec83c">
<div class="sourceCode cell-code" id="cb111"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb111-1"><a href="#cb111-1" aria-hidden="true" tabindex="-1"></a>ms_top <span class="ot">&lt;-</span> <span class="fu">subset</span>(dat, id.exposure <span class="sc">==</span> <span class="st">"ieu-b-18"</span> <span class="sc">&amp;</span> id.outcome <span class="sc">%in%</span> <span class="fu">unique</span>(<span class="fu">subset</span>(prs_pairs, opengwasid<span class="sc">==</span><span class="st">"ieu-b-18"</span>))<span class="sc">$</span>prot[<span class="dv">1</span><span class="sc">:</span><span class="dv">56</span>])</span>
<span id="cb111-2"><a href="#cb111-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-3"><a href="#cb111-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(ms_top, <span class="fu">aes</span>(<span class="at">x=</span>beta.exposure, <span class="at">y=</span>beta.outcome)) <span class="sc">+</span></span>
<span id="cb111-4"><a href="#cb111-4" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb111-5"><a href="#cb111-5" aria-hidden="true" tabindex="-1"></a><span class="fu">facet_wrap</span>(<span class="sc">~</span>id.outcome, <span class="at">scale=</span><span class="st">"free_y"</span>) <span class="sc">+</span></span>
<span id="cb111-6"><a href="#cb111-6" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_smooth</span>(<span class="at">method=</span><span class="st">"lm"</span>) <span class="sc">+</span></span>
<span id="cb111-7"><a href="#cb111-7" aria-hidden="true" tabindex="-1"></a><span class="fu">theme</span>(<span class="at">axis.text.x=</span><span class="fu">element_blank</span>(), <span class="at">axis.text.y=</span><span class="fu">element_blank</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`geom_smooth()` using formula = 'y ~ x'</code></pre>
</div>
<div class="cell-output-display">
<p><img src="reverse_mr_files/figure-html/unnamed-chunk-28-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell" data-hash="reverse_mr_cache/html/unnamed-chunk-29_8ff8e363e76ae4fb51d9c760031e6fe5">
<div class="sourceCode cell-code" id="cb113"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb113-1"><a href="#cb113-1" aria-hidden="true" tabindex="-1"></a>plot_rev_mr <span class="ot">&lt;-</span> <span class="cf">function</span>(id, nplot) {</span>
<span id="cb113-2"><a href="#cb113-2" aria-hidden="true" tabindex="-1"></a>    ms_top <span class="ot">&lt;-</span> <span class="fu">subset</span>(dat, id.exposure <span class="sc">%in%</span> id <span class="sc">&amp;</span> id.outcome <span class="sc">%in%</span> <span class="fu">unique</span>(<span class="fu">subset</span>(prs_pairs, opengwasid <span class="sc">%in%</span> id))<span class="sc">$</span>prot[<span class="dv">1</span><span class="sc">:</span>nplot])</span>
<span id="cb113-3"><a href="#cb113-3" aria-hidden="true" tabindex="-1"></a>    ind <span class="ot">&lt;-</span> <span class="fu">sign</span>(ms_top<span class="sc">$</span>beta.exposure) <span class="sc">==</span> <span class="sc">-</span><span class="dv">1</span></span>
<span id="cb113-4"><a href="#cb113-4" aria-hidden="true" tabindex="-1"></a>    ms_top<span class="sc">$</span>beta.exposure <span class="ot">&lt;-</span> <span class="fu">abs</span>(ms_top<span class="sc">$</span>beta.exposure)</span>
<span id="cb113-5"><a href="#cb113-5" aria-hidden="true" tabindex="-1"></a>    ms_top<span class="sc">$</span>beta.outcome[ind] <span class="ot">&lt;-</span> ms_top<span class="sc">$</span>beta.outcome[ind] <span class="sc">*</span> <span class="sc">-</span><span class="dv">1</span></span>
<span id="cb113-6"><a href="#cb113-6" aria-hidden="true" tabindex="-1"></a>    ms_top <span class="ot">&lt;-</span> <span class="fu">subset</span>(ms_top, <span class="sc">!</span> (chr.outcome <span class="sc">==</span> <span class="dv">6</span> <span class="sc">&amp;</span> pos.outcome <span class="sc">&gt;</span> <span class="dv">28000000</span> <span class="sc">&amp;</span> pos.outcome <span class="sc">&lt;</span> <span class="dv">33000000</span>))</span>
<span id="cb113-7"><a href="#cb113-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-8"><a href="#cb113-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(ms_top, <span class="fu">aes</span>(<span class="at">x=</span>beta.exposure, <span class="at">y=</span>beta.outcome)) <span class="sc">+</span></span>
<span id="cb113-9"><a href="#cb113-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb113-10"><a href="#cb113-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span>id.outcome, <span class="at">scale=</span><span class="st">"free_y"</span>) <span class="sc">+</span></span>
<span id="cb113-11"><a href="#cb113-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_smooth</span>(<span class="at">method=</span><span class="st">"lm"</span>) <span class="sc">+</span></span>
<span id="cb113-12"><a href="#cb113-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x=</span><span class="fu">element_blank</span>(), <span class="at">axis.text.y=</span><span class="fu">element_blank</span>())</span>
<span id="cb113-13"><a href="#cb113-13" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb113-14"><a href="#cb113-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-15"><a href="#cb113-15" aria-hidden="true" tabindex="-1"></a><span class="co"># plot_rev_mr("ebi-a-GCST003374", 56)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>MS excluding chr 6</p>
<div class="cell" data-hash="reverse_mr_cache/html/unnamed-chunk-30_68e27a22aeaef7c014f174c2b9b644b7">
<div class="sourceCode cell-code" id="cb114"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb114-1"><a href="#cb114-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_rev_mr</span>(<span class="fu">c</span>(<span class="st">"ieu-a-1024"</span>, <span class="st">"ieu-b-18"</span>), <span class="dv">56</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`geom_smooth()` using formula = 'y ~ x'</code></pre>
</div>
<div class="cell-output-display">
<p><img src="reverse_mr_files/figure-html/unnamed-chunk-30-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell" data-hash="reverse_mr_cache/html/unnamed-chunk-31_6a68091cd06b86139474738fbd30b5a5">
<div class="sourceCode cell-code" id="cb116"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb116-1"><a href="#cb116-1" aria-hidden="true" tabindex="-1"></a><span class="fu">subset</span>(dat, exposure <span class="sc">==</span> <span class="st">"MS"</span>) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(outcome <span class="sc">==</span> outcome[<span class="dv">1</span>])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] SNP                    effect_allele.exposure other_allele.exposure 
 [4] effect_allele.outcome  other_allele.outcome   beta.exposure         
 [7] beta.outcome           eaf.exposure           eaf.outcome           
[10] remove                 palindromic            ambiguous             
[13] id.outcome             chr.outcome            pos.outcome           
[16] se.outcome             outcome                mr_keep.outcome       
[19] pval.outcome           pval_origin.outcome    se.exposure           
[22] chr.exposure           pval.exposure          exposure              
[25] mr_keep.exposure       pval_origin.exposure   id.exposure           
[28] action                 mr_keep                samplesize.outcome    
&lt;0 rows&gt; (or 0-length row.names)</code></pre>
</div>
</div>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "î§‹";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>